// angular-google-charts/esm2015/lib/google-charts.module.js
import { NgModule } from '@angular/core';
import { ChartEditorComponent } from './components/chart-editor/chart-editor.component';
import { ChartWrapperComponent } from './components/chart-wrapper/chart-wrapper.component';
import { ControlWrapperComponent } from './components/control-wrapper/control-wrapper.component';
import { DashboardComponent } from './components/dashboard/dashboard.component';
import { GoogleChartComponent } from './components/google-chart/google-chart.component';
import { ScriptLoaderService } from './services/script-loader.service';
import { GOOGLE_CHARTS_CONFIG } from './types/google-charts-config';
import * as ɵngcc0 from '@angular/core';
export class GoogleChartsModule {
    static forRoot(config = {}) {
        return {
            ngModule: GoogleChartsModule,
            providers: [{ provide: GOOGLE_CHARTS_CONFIG, useValue: config }]
        };
    }
}
GoogleChartsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: GoogleChartsModule });
GoogleChartsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function GoogleChartsModule_Factory(t) { return new (t || GoogleChartsModule)(); }, providers: [ScriptLoaderService] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(GoogleChartsModule, { declarations: function () { return [GoogleChartComponent,
        ChartWrapperComponent,
        DashboardComponent,
        ControlWrapperComponent,
        ChartEditorComponent]; }, exports: function () { return [GoogleChartComponent,
        ChartWrapperComponent,
        DashboardComponent,
        ControlWrapperComponent,
        ChartEditorComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GoogleChartsModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    GoogleChartComponent,
                    ChartWrapperComponent,
                    DashboardComponent,
                    ControlWrapperComponent,
                    ChartEditorComponent
                ],
                providers: [ScriptLoaderService],
                exports: [
                    GoogleChartComponent,
                    ChartWrapperComponent,
                    DashboardComponent,
                    ControlWrapperComponent,
                    ChartEditorComponent
                ]
            }]
    }], null, null); })();
