// angular-pipes/esm2015/array/count.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { count } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let CountPipe = class CountPipe {
    transform(input) {
        return count(input);
    }
};
CountPipe.ɵfac = function CountPipe_Factory(t) { return new (t || CountPipe)(); };
CountPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "count", type: CountPipe, pure: true });
export { CountPipe };
let NgCountPipeModule = class NgCountPipeModule {
};
NgCountPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgCountPipeModule });
NgCountPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgCountPipeModule_Factory(t) { return new (t || NgCountPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CountPipe, [{
        type: Pipe,
        args: [{
                name: 'count'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgCountPipeModule, { declarations: [CountPipe], exports: [CountPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgCountPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [CountPipe],
                exports: [CountPipe]
            }]
    }], null, null); })();
export { NgCountPipeModule };
