// angular-pipes/esm2015/array/deep.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { wrapDeep, isDeepObject } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let DeepPipe = class DeepPipe {
    transform(value) {
        if (isDeepObject(value)) {
            return value;
        }
        return wrapDeep(value);
    }
};
DeepPipe.ɵfac = function DeepPipe_Factory(t) { return new (t || DeepPipe)(); };
DeepPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "deep", type: DeepPipe, pure: true });
export { DeepPipe };
let NgDeepPipeModule = class NgDeepPipeModule {
};
NgDeepPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDeepPipeModule });
NgDeepPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDeepPipeModule_Factory(t) { return new (t || NgDeepPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DeepPipe, [{
        type: Pipe,
        args: [{
                name: 'deep'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDeepPipeModule, { declarations: [DeepPipe], exports: [DeepPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDeepPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DeepPipe],
                exports: [DeepPipe]
            }]
    }], null, null); })();
export { NgDeepPipeModule };
