// angular-pipes/esm2015/array/empty.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { empty } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let EmptyPipe = class EmptyPipe {
    transform(input) {
        return empty(input);
    }
};
EmptyPipe.ɵfac = function EmptyPipe_Factory(t) { return new (t || EmptyPipe)(); };
EmptyPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "empty", type: EmptyPipe, pure: true });
export { EmptyPipe };
let NgEmptyPipeModule = class NgEmptyPipeModule {
};
NgEmptyPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEmptyPipeModule });
NgEmptyPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEmptyPipeModule_Factory(t) { return new (t || NgEmptyPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmptyPipe, [{
        type: Pipe,
        args: [{
                name: 'empty'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEmptyPipeModule, { declarations: [EmptyPipe], exports: [EmptyPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEmptyPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EmptyPipe],
                exports: [EmptyPipe]
            }]
    }], null, null); })();
export { NgEmptyPipeModule };
