// angular-pipes/esm2015/array/flatten.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, flatten } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let FlattenPipe = class FlattenPipe {
    transform(input) {
        const isDeep = isDeepObject(input);
        if (!isArray(input) && !isDeep) {
            return input;
        }
        if (isDeep) {
            const unwrapped = unwrapDeep(input);
            if (!isArray(unwrapped)) {
                return unwrapped;
            }
            return flatten(unwrapped, 0);
        }
        return [].concat.apply([], input);
    }
};
FlattenPipe.ɵfac = function FlattenPipe_Factory(t) { return new (t || FlattenPipe)(); };
FlattenPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "flatten", type: FlattenPipe, pure: true });
export { FlattenPipe };
let NgFlattenPipeModule = class NgFlattenPipeModule {
};
NgFlattenPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgFlattenPipeModule });
NgFlattenPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgFlattenPipeModule_Factory(t) { return new (t || NgFlattenPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlattenPipe, [{
        type: Pipe,
        args: [{ name: 'flatten' }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgFlattenPipeModule, { declarations: [FlattenPipe], exports: [FlattenPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgFlattenPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [FlattenPipe],
                exports: [FlattenPipe]
            }]
    }], null, null); })();
export { NgFlattenPipeModule };
