// angular-pipes/esm2015/array/head.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let HeadPipe = class HeadPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        // Will return undefined if length is 0
        return input[0];
    }
};
HeadPipe.ɵfac = function HeadPipe_Factory(t) { return new (t || HeadPipe)(); };
HeadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "head", type: HeadPipe, pure: true });
export { HeadPipe };
let NgHeadPipeModule = class NgHeadPipeModule {
};
NgHeadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgHeadPipeModule });
NgHeadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgHeadPipeModule_Factory(t) { return new (t || NgHeadPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HeadPipe, [{
        type: Pipe,
        args: [{
                name: 'head'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgHeadPipeModule, { declarations: [HeadPipe], exports: [HeadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgHeadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [HeadPipe],
                exports: [HeadPipe]
            }]
    }], null, null); })();
export { NgHeadPipeModule };
