// angular-pipes/esm2015/array/intersection.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IntersectionPipe = class IntersectionPipe {
    transform(a, b) {
        if ((!isArray(a) && !isDeepObject(a)) || !isArray(b)) {
            return [];
        }
        if (isDeepObject(a)) {
            const unwrapped = unwrapDeep(a);
            if (!isArray(unwrapped)) {
                return [];
            }
            return unwrapped.reduce((intersection, value) => intersection.concat(deepIndexOf(b, value) !== -1 && deepIndexOf(intersection, value) === -1 ? value : []), []);
        }
        return a.reduce((intersection, value) => intersection.concat(b.indexOf(value) !== -1 && intersection.indexOf(value) === -1 ? value : []), []);
    }
};
IntersectionPipe.ɵfac = function IntersectionPipe_Factory(t) { return new (t || IntersectionPipe)(); };
IntersectionPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "intersection", type: IntersectionPipe, pure: true });
export { IntersectionPipe };
let NgIntersectionPipeModule = class NgIntersectionPipeModule {
};
NgIntersectionPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIntersectionPipeModule });
NgIntersectionPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIntersectionPipeModule_Factory(t) { return new (t || NgIntersectionPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IntersectionPipe, [{
        type: Pipe,
        args: [{
                name: 'intersection'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIntersectionPipeModule, { declarations: [IntersectionPipe], exports: [IntersectionPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIntersectionPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IntersectionPipe],
                exports: [IntersectionPipe]
            }]
    }], null, null); })();
export { NgIntersectionPipeModule };
