// angular-pipes/esm2015/array/join.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let JoinPipe = class JoinPipe {
    transform(input, character = '') {
        if (!isArray(input)) {
            return input;
        }
        return input.join(character);
    }
};
JoinPipe.ɵfac = function JoinPipe_Factory(t) { return new (t || JoinPipe)(); };
JoinPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "join", type: JoinPipe, pure: true });
export { JoinPipe };
let NgJoinPipeModule = class NgJoinPipeModule {
};
NgJoinPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgJoinPipeModule });
NgJoinPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgJoinPipeModule_Factory(t) { return new (t || NgJoinPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(JoinPipe, [{
        type: Pipe,
        args: [{
                name: 'join'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgJoinPipeModule, { declarations: [JoinPipe], exports: [JoinPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgJoinPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [JoinPipe],
                exports: [JoinPipe]
            }]
    }], null, null); })();
export { NgJoinPipeModule };
