// angular-pipes/esm2015/array/last.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let LastPipe = class LastPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        // Returns undefined if empty
        return input[input.length - 1];
    }
};
LastPipe.ɵfac = function LastPipe_Factory(t) { return new (t || LastPipe)(); };
LastPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "last", type: LastPipe, pure: true });
export { LastPipe };
let NgLastPipeModule = class NgLastPipeModule {
};
NgLastPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgLastPipeModule });
NgLastPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgLastPipeModule_Factory(t) { return new (t || NgLastPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LastPipe, [{
        type: Pipe,
        args: [{
                name: 'last'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgLastPipeModule, { declarations: [LastPipe], exports: [LastPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgLastPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [LastPipe],
                exports: [LastPipe]
            }]
    }], null, null); })();
export { NgLastPipeModule };
