// angular-pipes/esm2015/array/range.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let RangePipe = class RangePipe {
    transform(_input, size = 0, start = 1, step = 1) {
        const range = [];
        for (let length = 0; length < size; ++length) {
            range.push(start);
            start += step;
        }
        return range;
    }
};
RangePipe.ɵfac = function RangePipe_Factory(t) { return new (t || RangePipe)(); };
RangePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "range", type: RangePipe, pure: true });
export { RangePipe };
let NgRangePipeModule = class NgRangePipeModule {
};
NgRangePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRangePipeModule });
NgRangePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRangePipeModule_Factory(t) { return new (t || NgRangePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RangePipe, [{
        type: Pipe,
        args: [{
                name: 'range'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRangePipeModule, { declarations: [RangePipe], exports: [RangePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRangePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RangePipe],
                exports: [RangePipe]
            }]
    }], null, null); })();
export { NgRangePipeModule };
