// angular-pipes/esm2015/array/some.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let SomePipe = class SomePipe {
    transform(input, predicate) {
        if (!isArray(input) || !predicate) {
            return input;
        }
        let result = false;
        let i = -1;
        while (++i < input.length && !result) {
            result = predicate(input[i], i, input);
        }
        return result;
    }
};
SomePipe.ɵfac = function SomePipe_Factory(t) { return new (t || SomePipe)(); };
SomePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "some", type: SomePipe, pure: true });
export { SomePipe };
let NgSomePipeModule = class NgSomePipeModule {
};
NgSomePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSomePipeModule });
NgSomePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSomePipeModule_Factory(t) { return new (t || NgSomePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SomePipe, [{
        type: Pipe,
        args: [{
                name: 'some'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSomePipeModule, { declarations: [SomePipe], exports: [SomePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSomePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SomePipe],
                exports: [SomePipe]
            }]
    }], null, null); })();
export { NgSomePipeModule };
