// angular-pipes/esm2015/array/tail.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TailPipe = class TailPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        return input.slice(1, input.length);
    }
};
TailPipe.ɵfac = function TailPipe_Factory(t) { return new (t || TailPipe)(); };
TailPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "tail", type: TailPipe, pure: true });
export { TailPipe };
let NgTailPipeModule = class NgTailPipeModule {
};
NgTailPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTailPipeModule });
NgTailPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTailPipeModule_Factory(t) { return new (t || NgTailPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TailPipe, [{
        type: Pipe,
        args: [{
                name: 'tail'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTailPipeModule, { declarations: [TailPipe], exports: [TailPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTailPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TailPipe],
                exports: [TailPipe]
            }]
    }], null, null); })();
export { NgTailPipeModule };
