// angular-pipes/esm2015/array/take-while.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, takeWhile, isNil } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TakeWhilePipe = class TakeWhilePipe {
    transform(input, predicate) {
        if (!isArray(input) || isNil(predicate)) {
            return input;
        }
        return takeWhile(input, predicate);
    }
};
TakeWhilePipe.ɵfac = function TakeWhilePipe_Factory(t) { return new (t || TakeWhilePipe)(); };
TakeWhilePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "takeWhile", type: TakeWhilePipe, pure: true });
export { TakeWhilePipe };
let NgTakeWhilePipeModule = class NgTakeWhilePipeModule {
};
NgTakeWhilePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTakeWhilePipeModule });
NgTakeWhilePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTakeWhilePipeModule_Factory(t) { return new (t || NgTakeWhilePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TakeWhilePipe, [{
        type: Pipe,
        args: [{
                name: 'takeWhile'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTakeWhilePipeModule, { declarations: [TakeWhilePipe], exports: [TakeWhilePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTakeWhilePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TakeWhilePipe],
                exports: [TakeWhilePipe]
            }]
    }], null, null); })();
export { NgTakeWhilePipeModule };
