// angular-pipes/esm2015/array/take.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TakePipe = class TakePipe {
    transform(input, quantity) {
        if (!isArray(input)) {
            return input;
        }
        return input.slice(0, quantity || 1);
    }
};
TakePipe.ɵfac = function TakePipe_Factory(t) { return new (t || TakePipe)(); };
TakePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "take", type: TakePipe, pure: true });
export { TakePipe };
let NgTakePipeModule = class NgTakePipeModule {
};
NgTakePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTakePipeModule });
NgTakePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTakePipeModule_Factory(t) { return new (t || NgTakePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TakePipe, [{
        type: Pipe,
        args: [{
                name: 'take'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTakePipeModule, { declarations: [TakePipe], exports: [TakePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTakePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TakePipe],
                exports: [TakePipe]
            }]
    }], null, null); })();
export { NgTakePipeModule };
