// angular-pipes/esm2015/array/take-until.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, takeUntil, isNil } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TakeUntilPipe = class TakeUntilPipe {
    transform(input, predicate) {
        if (!isArray(input) || isNil(predicate)) {
            return input;
        }
        return takeUntil(input, predicate);
    }
};
TakeUntilPipe.ɵfac = function TakeUntilPipe_Factory(t) { return new (t || TakeUntilPipe)(); };
TakeUntilPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "takeUntil", type: TakeUntilPipe, pure: true });
export { TakeUntilPipe };
let NgTakeUntilPipeModule = class NgTakeUntilPipeModule {
};
NgTakeUntilPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTakeUntilPipeModule });
NgTakeUntilPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTakeUntilPipeModule_Factory(t) { return new (t || NgTakeUntilPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TakeUntilPipe, [{
        type: Pipe,
        args: [{
                name: 'takeUntil'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTakeUntilPipeModule, { declarations: [TakeUntilPipe], exports: [TakeUntilPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTakeUntilPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TakeUntilPipe],
                exports: [TakeUntilPipe]
            }]
    }], null, null); })();
export { NgTakeUntilPipeModule };
