// angular-pipes/esm2015/array/union.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let UnionPipe = class UnionPipe {
    transform(a, b) {
        if ((!isArray(a) && !isDeepObject(a)) || !isArray(b)) {
            return [];
        }
        if (isDeepObject(a)) {
            const unwrapped = unwrapDeep(a);
            if (!isArray(unwrapped)) {
                return [];
            }
            return []
                .concat(unwrapped)
                .concat(b)
                .filter((value, index, input) => deepIndexOf(input, value) === index);
        }
        return []
            .concat(a)
            .concat(b)
            .filter((value, index, input) => input.indexOf(value) === index);
    }
};
UnionPipe.ɵfac = function UnionPipe_Factory(t) { return new (t || UnionPipe)(); };
UnionPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "union", type: UnionPipe, pure: true });
export { UnionPipe };
let NgUnionPipeModule = class NgUnionPipeModule {
};
NgUnionPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUnionPipeModule });
NgUnionPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUnionPipeModule_Factory(t) { return new (t || NgUnionPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnionPipe, [{
        type: Pipe,
        args: [{
                name: 'union'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUnionPipeModule, { declarations: [UnionPipe], exports: [UnionPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUnionPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [UnionPipe],
                exports: [UnionPipe]
            }]
    }], null, null); })();
export { NgUnionPipeModule };
