// angular-pipes/esm2015/array/uniq.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let UniqPipe = class UniqPipe {
    transform(input) {
        if (!isArray(input) && !isDeepObject(input)) {
            return input;
        }
        if (isDeepObject(input)) {
            const unwrappedInput = unwrapDeep(input);
            if (!isArray(unwrappedInput)) {
                return unwrappedInput;
            }
            return unwrappedInput.filter((value, index) => deepIndexOf(unwrappedInput, value) === index);
        }
        return input.filter((value, index) => input.indexOf(value) === index);
    }
};
UniqPipe.ɵfac = function UniqPipe_Factory(t) { return new (t || UniqPipe)(); };
UniqPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "uniq", type: UniqPipe, pure: true });
export { UniqPipe };
let NgUniqPipeModule = class NgUniqPipeModule {
};
NgUniqPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUniqPipeModule });
NgUniqPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUniqPipeModule_Factory(t) { return new (t || NgUniqPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UniqPipe, [{
        type: Pipe,
        args: [{
                name: 'uniq'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUniqPipeModule, { declarations: [UniqPipe], exports: [UniqPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUniqPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [UniqPipe],
                exports: [UniqPipe]
            }]
    }], null, null); })();
export { NgUniqPipeModule };
