// angular-pipes/esm2015/array/where.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isFunction, getProperty } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let WherePipe = class WherePipe {
    /**
     * Support a function or a value or the shorthand ['key', value] like the lodash shorthand.
     */
    transform(input, fn) {
        if (!isArray(input)) {
            return input;
        }
        if (isFunction(fn)) {
            return input.filter(fn);
        }
        else if (isArray(fn)) {
            const [key, value] = fn;
            return input.filter((item) => getProperty(item, key) === value);
        }
        else if (fn) {
            return input.filter((item) => item === fn);
        }
        else {
            return input;
        }
    }
};
WherePipe.ɵfac = function WherePipe_Factory(t) { return new (t || WherePipe)(); };
WherePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "where", type: WherePipe, pure: true });
export { WherePipe };
let NgWherePipeModule = class NgWherePipeModule {
};
NgWherePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgWherePipeModule });
NgWherePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgWherePipeModule_Factory(t) { return new (t || NgWherePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WherePipe, [{
        type: Pipe,
        args: [{
                name: 'where'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgWherePipeModule, { declarations: [WherePipe], exports: [WherePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgWherePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [WherePipe],
                exports: [WherePipe]
            }]
    }], null, null); })();
export { NgWherePipeModule };
