// angular-pipes/esm2015/array/without.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let WithoutPipe = class WithoutPipe {
    transform(input, ...args) {
        if (!isArray(input) && !isDeepObject(input)) {
            return input;
        }
        if (isDeepObject(input)) {
            const unwrappedInput = unwrapDeep(input);
            if (!isArray(unwrappedInput)) {
                return unwrappedInput;
            }
            return unwrappedInput.filter((value) => deepIndexOf(args, value) === -1);
        }
        return input.filter((value) => args.indexOf(value) === -1);
    }
};
WithoutPipe.ɵfac = function WithoutPipe_Factory(t) { return new (t || WithoutPipe)(); };
WithoutPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "without", type: WithoutPipe, pure: true });
export { WithoutPipe };
let NgWithoutPipeModule = class NgWithoutPipeModule {
};
NgWithoutPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgWithoutPipeModule });
NgWithoutPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgWithoutPipeModule_Factory(t) { return new (t || NgWithoutPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WithoutPipe, [{
        type: Pipe,
        args: [{
                name: 'without'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgWithoutPipeModule, { declarations: [WithoutPipe], exports: [WithoutPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgWithoutPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [WithoutPipe],
                exports: [WithoutPipe]
            }]
    }], null, null); })();
export { NgWithoutPipeModule };
