// angular-pipes/esm2015/object/to-array.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isObject } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let ToArrayPipe = class ToArrayPipe {
    transform(input) {
        if (!isObject(input)) {
            return input;
        }
        return Object.keys(input).map(value => input[value]);
    }
};
ToArrayPipe.ɵfac = function ToArrayPipe_Factory(t) { return new (t || ToArrayPipe)(); };
ToArrayPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "toArray", type: ToArrayPipe, pure: true });
export { ToArrayPipe };
let NgToArrayPipeModule = class NgToArrayPipeModule {
};
NgToArrayPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgToArrayPipeModule });
NgToArrayPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgToArrayPipeModule_Factory(t) { return new (t || NgToArrayPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ToArrayPipe, [{
        type: Pipe,
        args: [{ name: 'toArray' }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgToArrayPipeModule, { declarations: [ToArrayPipe], exports: [ToArrayPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgToArrayPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ToArrayPipe],
                exports: [ToArrayPipe]
            }]
    }], null, null); })();
export { NgToArrayPipeModule };
