// angular-pipes/esm5/boolean/boolean.module.js
import * as tslib_1 from "tslib";
import { NgModule } from '@angular/core';
import { NgIsEqualPipeModule } from './is-equal.pipe';
import { NgIsGreaterOrEqualPipeModule } from './is-greater-or-equal.pipe';
import { NgIsGreaterPipeModule } from './is-greater.pipe';
import { NgIsIdenticalPipeModule } from './is-identical.pipe';
import { NgIsLessOrEqualPipeModule } from './is-less-or-equal.pipe';
import { NgIsLessPipeModule } from './is-less.pipe';
import { NgIsNotEqualPipeModule } from './is-not-equal.pipe';
import { NgIsNotIdenticalPipeModule } from './is-not-identical.pipe';
import { NgIsNullPipeModule } from './is-null.pipe';
import { NgIsUndefinedPipeModule } from './is-undefined.pipe';
import { NgIsNilPipeModule } from './is-nil.pipe';
import { NgIsFunctionPipeModule } from './is-function.pipe';
import { NgIsNumberPipeModule } from './is-number.pipe';
import { NgIsStringPipeModule } from './is-string.pipe';
import { NgIsArrayPipeModule } from './is-array.pipe';
import { NgIsObjectPipeModule } from './is-object.pipe';
import { NgIsDefinedPipeModule } from './is-defined.pipe';
import * as ɵngcc0 from '@angular/core';
var NgTypesPipesModule = /** @class */ (function () {
    function NgTypesPipesModule() {
    }
NgTypesPipesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTypesPipesModule });
NgTypesPipesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTypesPipesModule_Factory(t) { return new (t || NgTypesPipesModule)(); }, imports: [[
            NgIsNullPipeModule,
            NgIsUndefinedPipeModule,
            NgIsNilPipeModule,
            NgIsFunctionPipeModule,
            NgIsNumberPipeModule,
            NgIsStringPipeModule,
            NgIsArrayPipeModule,
            NgIsObjectPipeModule,
            NgIsDefinedPipeModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTypesPipesModule, { imports: function () { return [NgIsNullPipeModule,
        NgIsUndefinedPipeModule,
        NgIsNilPipeModule,
        NgIsFunctionPipeModule,
        NgIsNumberPipeModule,
        NgIsStringPipeModule,
        NgIsArrayPipeModule,
        NgIsObjectPipeModule,
        NgIsDefinedPipeModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTypesPipesModule, [{
        type: NgModule,
        args: [{
                imports: [
                    NgIsNullPipeModule,
                    NgIsUndefinedPipeModule,
                    NgIsNilPipeModule,
                    NgIsFunctionPipeModule,
                    NgIsNumberPipeModule,
                    NgIsStringPipeModule,
                    NgIsArrayPipeModule,
                    NgIsObjectPipeModule,
                    NgIsDefinedPipeModule,
                ]
            }]
    }], function () { return []; }, null); })();
    return NgTypesPipesModule;
}());
export { NgTypesPipesModule };
var NgConditionsPipesModule = /** @class */ (function () {
    function NgConditionsPipesModule() {
    }
NgConditionsPipesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgConditionsPipesModule });
NgConditionsPipesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgConditionsPipesModule_Factory(t) { return new (t || NgConditionsPipesModule)(); }, imports: [[
            NgIsEqualPipeModule,
            NgIsGreaterOrEqualPipeModule,
            NgIsGreaterPipeModule,
            NgIsIdenticalPipeModule,
            NgIsLessOrEqualPipeModule,
            NgIsLessPipeModule,
            NgIsNotEqualPipeModule,
            NgIsNotIdenticalPipeModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgConditionsPipesModule, { imports: function () { return [NgIsEqualPipeModule,
        NgIsGreaterOrEqualPipeModule,
        NgIsGreaterPipeModule,
        NgIsIdenticalPipeModule,
        NgIsLessOrEqualPipeModule,
        NgIsLessPipeModule,
        NgIsNotEqualPipeModule,
        NgIsNotIdenticalPipeModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgConditionsPipesModule, [{
        type: NgModule,
        args: [{
                imports: [
                    NgIsEqualPipeModule,
                    NgIsGreaterOrEqualPipeModule,
                    NgIsGreaterPipeModule,
                    NgIsIdenticalPipeModule,
                    NgIsLessOrEqualPipeModule,
                    NgIsLessPipeModule,
                    NgIsNotEqualPipeModule,
                    NgIsNotIdenticalPipeModule,
                ]
            }]
    }], function () { return []; }, null); })();
    return NgConditionsPipesModule;
}());
export { NgConditionsPipesModule };
var NgBooleanPipesModule = /** @class */ (function () {
    function NgBooleanPipesModule() {
    }
NgBooleanPipesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgBooleanPipesModule });
NgBooleanPipesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgBooleanPipesModule_Factory(t) { return new (t || NgBooleanPipesModule)(); }, imports: [[NgConditionsPipesModule, NgTypesPipesModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgBooleanPipesModule, { imports: [NgConditionsPipesModule, NgTypesPipesModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgBooleanPipesModule, [{
        type: NgModule,
        args: [{
                imports: [NgConditionsPipesModule, NgTypesPipesModule]
            }]
    }], function () { return []; }, null); })();
    return NgBooleanPipesModule;
}());
export { NgBooleanPipesModule };
