// angular-pipes/esm5/boolean/is-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var IsEqualPipe = /** @class */ (function () {
    function IsEqualPipe() {
    }
    IsEqualPipe.prototype.transform = function (first, second) {
        return first == second;
    };
IsEqualPipe.ɵfac = function IsEqualPipe_Factory(t) { return new (t || IsEqualPipe)(); };
IsEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "equal", type: IsEqualPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'equal'
            }]
    }], function () { return []; }, null); })();
    return IsEqualPipe;
}());
export { IsEqualPipe };
var NgIsEqualPipeModule = /** @class */ (function () {
    function NgIsEqualPipeModule() {
    }
NgIsEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsEqualPipeModule });
NgIsEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsEqualPipeModule_Factory(t) { return new (t || NgIsEqualPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsEqualPipeModule, { declarations: [IsEqualPipe], exports: [IsEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsEqualPipe],
                exports: [IsEqualPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsEqualPipeModule;
}());
export { NgIsEqualPipeModule };
