// angular-pipes/esm5/boolean/is-function.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isFunction } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var IsFunctionPipe = /** @class */ (function () {
    function IsFunctionPipe() {
    }
    IsFunctionPipe.prototype.transform = function (value) {
        return isFunction(value);
    };
IsFunctionPipe.ɵfac = function IsFunctionPipe_Factory(t) { return new (t || IsFunctionPipe)(); };
IsFunctionPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isFunction", type: IsFunctionPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsFunctionPipe, [{
        type: Pipe,
        args: [{
                name: 'isFunction'
            }]
    }], function () { return []; }, null); })();
    return IsFunctionPipe;
}());
export { IsFunctionPipe };
var NgIsFunctionPipeModule = /** @class */ (function () {
    function NgIsFunctionPipeModule() {
    }
NgIsFunctionPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsFunctionPipeModule });
NgIsFunctionPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsFunctionPipeModule_Factory(t) { return new (t || NgIsFunctionPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsFunctionPipeModule, { declarations: [IsFunctionPipe], exports: [IsFunctionPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsFunctionPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsFunctionPipe],
                exports: [IsFunctionPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsFunctionPipeModule;
}());
export { NgIsFunctionPipeModule };
