// angular-pipes/esm5/boolean/is-greater.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var IsGreaterPipe = /** @class */ (function () {
    function IsGreaterPipe() {
    }
    IsGreaterPipe.prototype.transform = function (first, second) {
        return first > second;
    };
IsGreaterPipe.ɵfac = function IsGreaterPipe_Factory(t) { return new (t || IsGreaterPipe)(); };
IsGreaterPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "greater", type: IsGreaterPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsGreaterPipe, [{
        type: Pipe,
        args: [{
                name: 'greater'
            }]
    }], function () { return []; }, null); })();
    return IsGreaterPipe;
}());
export { IsGreaterPipe };
var NgIsGreaterPipeModule = /** @class */ (function () {
    function NgIsGreaterPipeModule() {
    }
NgIsGreaterPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsGreaterPipeModule });
NgIsGreaterPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsGreaterPipeModule_Factory(t) { return new (t || NgIsGreaterPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsGreaterPipeModule, { declarations: [IsGreaterPipe], exports: [IsGreaterPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsGreaterPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsGreaterPipe],
                exports: [IsGreaterPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsGreaterPipeModule;
}());
export { NgIsGreaterPipeModule };
