// angular-pipes/esm5/boolean/is-less.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var IsLessPipe = /** @class */ (function () {
    function IsLessPipe() {
    }
    IsLessPipe.prototype.transform = function (first, second) {
        return first < second;
    };
IsLessPipe.ɵfac = function IsLessPipe_Factory(t) { return new (t || IsLessPipe)(); };
IsLessPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "less", type: IsLessPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsLessPipe, [{
        type: Pipe,
        args: [{
                name: 'less'
            }]
    }], function () { return []; }, null); })();
    return IsLessPipe;
}());
export { IsLessPipe };
var NgIsLessPipeModule = /** @class */ (function () {
    function NgIsLessPipeModule() {
    }
NgIsLessPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsLessPipeModule });
NgIsLessPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsLessPipeModule_Factory(t) { return new (t || NgIsLessPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsLessPipeModule, { declarations: [IsLessPipe], exports: [IsLessPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsLessPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsLessPipe],
                exports: [IsLessPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsLessPipeModule;
}());
export { NgIsLessPipeModule };
