// angular-pipes/esm5/boolean/is-not-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var IsNotEqualPipe = /** @class */ (function () {
    function IsNotEqualPipe() {
    }
    IsNotEqualPipe.prototype.transform = function (first, second) {
        return first != second;
    };
IsNotEqualPipe.ɵfac = function IsNotEqualPipe_Factory(t) { return new (t || IsNotEqualPipe)(); };
IsNotEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "notEqual", type: IsNotEqualPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNotEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'notEqual'
            }]
    }], function () { return []; }, null); })();
    return IsNotEqualPipe;
}());
export { IsNotEqualPipe };
var NgIsNotEqualPipeModule = /** @class */ (function () {
    function NgIsNotEqualPipeModule() {
    }
NgIsNotEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNotEqualPipeModule });
NgIsNotEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNotEqualPipeModule_Factory(t) { return new (t || NgIsNotEqualPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNotEqualPipeModule, { declarations: [IsNotEqualPipe], exports: [IsNotEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNotEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNotEqualPipe],
                exports: [IsNotEqualPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsNotEqualPipeModule;
}());
export { NgIsNotEqualPipeModule };
