// angular-pipes/esm5/boolean/is-number.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumber } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var IsNumberPipe = /** @class */ (function () {
    function IsNumberPipe() {
    }
    IsNumberPipe.prototype.transform = function (value) {
        return isNumber(value);
    };
IsNumberPipe.ɵfac = function IsNumberPipe_Factory(t) { return new (t || IsNumberPipe)(); };
IsNumberPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isNumber", type: IsNumberPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNumberPipe, [{
        type: Pipe,
        args: [{
                name: 'isNumber'
            }]
    }], function () { return []; }, null); })();
    return IsNumberPipe;
}());
export { IsNumberPipe };
var NgIsNumberPipeModule = /** @class */ (function () {
    function NgIsNumberPipeModule() {
    }
NgIsNumberPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNumberPipeModule });
NgIsNumberPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNumberPipeModule_Factory(t) { return new (t || NgIsNumberPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNumberPipeModule, { declarations: [IsNumberPipe], exports: [IsNumberPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNumberPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNumberPipe],
                exports: [IsNumberPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsNumberPipeModule;
}());
export { NgIsNumberPipeModule };
