// angular-pipes/esm5/boolean/is-nil.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNil } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var IsNilPipe = /** @class */ (function () {
    function IsNilPipe() {
    }
    IsNilPipe.prototype.transform = function (value) {
        return isNil(value);
    };
IsNilPipe.ɵfac = function IsNilPipe_Factory(t) { return new (t || IsNilPipe)(); };
IsNilPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isNil", type: IsNilPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNilPipe, [{
        type: Pipe,
        args: [{
                name: 'isNil'
            }]
    }], function () { return []; }, null); })();
    return IsNilPipe;
}());
export { IsNilPipe };
var NgIsNilPipeModule = /** @class */ (function () {
    function NgIsNilPipeModule() {
    }
NgIsNilPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNilPipeModule });
NgIsNilPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNilPipeModule_Factory(t) { return new (t || NgIsNilPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNilPipeModule, { declarations: [IsNilPipe], exports: [IsNilPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNilPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNilPipe],
                exports: [IsNilPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsNilPipeModule;
}());
export { NgIsNilPipeModule };
