// angular-pipes/esm5/boolean/is-object.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isObject } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var IsObjectPipe = /** @class */ (function () {
    function IsObjectPipe() {
    }
    IsObjectPipe.prototype.transform = function (value) {
        return isObject(value);
    };
IsObjectPipe.ɵfac = function IsObjectPipe_Factory(t) { return new (t || IsObjectPipe)(); };
IsObjectPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isObject", type: IsObjectPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsObjectPipe, [{
        type: Pipe,
        args: [{
                name: 'isObject'
            }]
    }], function () { return []; }, null); })();
    return IsObjectPipe;
}());
export { IsObjectPipe };
var NgIsObjectPipeModule = /** @class */ (function () {
    function NgIsObjectPipeModule() {
    }
NgIsObjectPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsObjectPipeModule });
NgIsObjectPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsObjectPipeModule_Factory(t) { return new (t || NgIsObjectPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsObjectPipeModule, { declarations: [IsObjectPipe], exports: [IsObjectPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsObjectPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsObjectPipe],
                exports: [IsObjectPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsObjectPipeModule;
}());
export { NgIsObjectPipeModule };
