// angular-pipes/esm5/boolean/is-string.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var IsStringPipe = /** @class */ (function () {
    function IsStringPipe() {
    }
    IsStringPipe.prototype.transform = function (value) {
        return isString(value);
    };
IsStringPipe.ɵfac = function IsStringPipe_Factory(t) { return new (t || IsStringPipe)(); };
IsStringPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isString", type: IsStringPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsStringPipe, [{
        type: Pipe,
        args: [{
                name: 'isString'
            }]
    }], function () { return []; }, null); })();
    return IsStringPipe;
}());
export { IsStringPipe };
var NgIsStringPipeModule = /** @class */ (function () {
    function NgIsStringPipeModule() {
    }
NgIsStringPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsStringPipeModule });
NgIsStringPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsStringPipeModule_Factory(t) { return new (t || NgIsStringPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsStringPipeModule, { declarations: [IsStringPipe], exports: [IsStringPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsStringPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsStringPipe],
                exports: [IsStringPipe]
            }]
    }], function () { return []; }, null); })();
    return NgIsStringPipeModule;
}());
export { NgIsStringPipeModule };
