// angular-pipes/esm2015/math/abs.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let AbsPipe = class AbsPipe {
    transform(input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return Math.abs(input);
    }
};
AbsPipe.ɵfac = function AbsPipe_Factory(t) { return new (t || AbsPipe)(); };
AbsPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "abs", type: AbsPipe, pure: true });
export { AbsPipe };
let NgAbsPipeModule = class NgAbsPipeModule {
};
NgAbsPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgAbsPipeModule });
NgAbsPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgAbsPipeModule_Factory(t) { return new (t || NgAbsPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AbsPipe, [{
        type: Pipe,
        args: [{
                name: 'abs'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgAbsPipeModule, { declarations: [AbsPipe], exports: [AbsPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgAbsPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [AbsPipe],
                exports: [AbsPipe]
            }]
    }], null, null); })();
export { NgAbsPipeModule };
