// angular-pipes/esm2015/math/degrees.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let DegreesPipe = class DegreesPipe {
    transform(input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return (input * 180) / Math.PI;
    }
};
DegreesPipe.ɵfac = function DegreesPipe_Factory(t) { return new (t || DegreesPipe)(); };
DegreesPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "degrees", type: DegreesPipe, pure: true });
export { DegreesPipe };
let NgDegreesPipeModule = class NgDegreesPipeModule {
};
NgDegreesPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDegreesPipeModule });
NgDegreesPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDegreesPipeModule_Factory(t) { return new (t || NgDegreesPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DegreesPipe, [{
        type: Pipe,
        args: [{
                name: 'degrees'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDegreesPipeModule, { declarations: [DegreesPipe], exports: [DegreesPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDegreesPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DegreesPipe],
                exports: [DegreesPipe]
            }]
    }], null, null); })();
export { NgDegreesPipeModule };
