// angular-pipes/esm2015/math/floor.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { createRound, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let FloorPipe = class FloorPipe {
    transform(value, precision = 0) {
        if (isString(precision)) {
            precision = parseInt(precision);
        }
        return createRound('floor')(value, precision);
    }
};
FloorPipe.ɵfac = function FloorPipe_Factory(t) { return new (t || FloorPipe)(); };
FloorPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "floor", type: FloorPipe, pure: true });
export { FloorPipe };
let NgFloorPipeModule = class NgFloorPipeModule {
};
NgFloorPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgFloorPipeModule });
NgFloorPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgFloorPipeModule_Factory(t) { return new (t || NgFloorPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FloorPipe, [{
        type: Pipe,
        args: [{
                name: 'floor'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgFloorPipeModule, { declarations: [FloorPipe], exports: [FloorPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgFloorPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [FloorPipe],
                exports: [FloorPipe]
            }]
    }], null, null); })();
export { NgFloorPipeModule };
