// angular-pipes/esm2015/math/ordinal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let OrdinalPipe = class OrdinalPipe {
    transform(input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        if (this.endsWithTenth(input)) {
            return input + 'th';
        }
        else {
            const cardinal = input.toString().charAt(input.toString().length - 1);
            switch (cardinal) {
                case '1':
                    return input + 'st';
                case '2':
                    return input + 'nd';
                case '3':
                    return input + 'rd';
                default:
                    return input + 'th';
            }
        }
    }
    endsWithTenth(input) {
        const beforeLastDigit = input.toString().charAt(input.toString().length - 2);
        return beforeLastDigit === '1';
    }
};
OrdinalPipe.ɵfac = function OrdinalPipe_Factory(t) { return new (t || OrdinalPipe)(); };
OrdinalPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "ordinal", type: OrdinalPipe, pure: true });
export { OrdinalPipe };
let NgOrdinalPipeModule = class NgOrdinalPipeModule {
};
NgOrdinalPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgOrdinalPipeModule });
NgOrdinalPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgOrdinalPipeModule_Factory(t) { return new (t || NgOrdinalPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OrdinalPipe, [{
        type: Pipe,
        args: [{
                name: 'ordinal'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgOrdinalPipeModule, { declarations: [OrdinalPipe], exports: [OrdinalPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgOrdinalPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [OrdinalPipe],
                exports: [OrdinalPipe]
            }]
    }], null, null); })();
export { NgOrdinalPipeModule };
