// angular-pipes/esm2015/math/pow.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let PowPipe = class PowPipe {
    transform(input, power = 2) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return Math.pow(input, power);
    }
};
PowPipe.ɵfac = function PowPipe_Factory(t) { return new (t || PowPipe)(); };
PowPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "pow", type: PowPipe, pure: true });
export { PowPipe };
let NgPowPipeModule = class NgPowPipeModule {
};
NgPowPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgPowPipeModule });
NgPowPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgPowPipeModule_Factory(t) { return new (t || NgPowPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PowPipe, [{
        type: Pipe,
        args: [{
                name: 'pow'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgPowPipeModule, { declarations: [PowPipe], exports: [PowPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgPowPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [PowPipe],
                exports: [PowPipe]
            }]
    }], null, null); })();
export { NgPowPipeModule };
