// angular-pipes/esm2015/math/radians.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let RadiansPipe = class RadiansPipe {
    transform(input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return (input * Math.PI) / 180;
    }
};
RadiansPipe.ɵfac = function RadiansPipe_Factory(t) { return new (t || RadiansPipe)(); };
RadiansPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "radians", type: RadiansPipe, pure: true });
export { RadiansPipe };
let NgRadiansPipeModule = class NgRadiansPipeModule {
};
NgRadiansPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRadiansPipeModule });
NgRadiansPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRadiansPipeModule_Factory(t) { return new (t || NgRadiansPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RadiansPipe, [{
        type: Pipe,
        args: [{
                name: 'radians'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRadiansPipeModule, { declarations: [RadiansPipe], exports: [RadiansPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRadiansPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RadiansPipe],
                exports: [RadiansPipe]
            }]
    }], null, null); })();
export { NgRadiansPipeModule };
