// angular-pipes/esm2015/math/random.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let RandomPipe = class RandomPipe {
    transform(input, min = 0, max = 1) {
        if (!isNumberFinite(min) || !isNumberFinite(max)) {
            return input;
        }
        if (min > max) {
            max = min;
            min = 0;
        }
        return Math.random() * (max - min) + min;
    }
};
RandomPipe.ɵfac = function RandomPipe_Factory(t) { return new (t || RandomPipe)(); };
RandomPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "random", type: RandomPipe, pure: true });
export { RandomPipe };
let NgRandomPipeModule = class NgRandomPipeModule {
};
NgRandomPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRandomPipeModule });
NgRandomPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRandomPipeModule_Factory(t) { return new (t || NgRandomPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RandomPipe, [{
        type: Pipe,
        args: [{
                name: 'random'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRandomPipeModule, { declarations: [RandomPipe], exports: [RandomPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRandomPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RandomPipe],
                exports: [RandomPipe]
            }]
    }], null, null); })();
export { NgRandomPipeModule };
