// angular-pipes/esm2015/math/round.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { createRound, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let RoundPipe = class RoundPipe {
    transform(value, precision = 0) {
        if (isString(precision)) {
            precision = parseInt(precision);
        }
        return createRound('round')(value, precision);
    }
};
RoundPipe.ɵfac = function RoundPipe_Factory(t) { return new (t || RoundPipe)(); };
RoundPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "round", type: RoundPipe, pure: true });
export { RoundPipe };
let NgRoundPipeModule = class NgRoundPipeModule {
};
NgRoundPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRoundPipeModule });
NgRoundPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRoundPipeModule_Factory(t) { return new (t || NgRoundPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RoundPipe, [{
        type: Pipe,
        args: [{
                name: 'round'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRoundPipeModule, { declarations: [RoundPipe], exports: [RoundPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRoundPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RoundPipe],
                exports: [RoundPipe]
            }]
    }], null, null); })();
export { NgRoundPipeModule };
