// angular-pipes/esm2015/math/sqrt.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let SqrtPipe = class SqrtPipe {
    transform(input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return Math.sqrt(input);
    }
};
SqrtPipe.ɵfac = function SqrtPipe_Factory(t) { return new (t || SqrtPipe)(); };
SqrtPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "sqrt", type: SqrtPipe, pure: true });
export { SqrtPipe };
let NgSqrtPipeModule = class NgSqrtPipeModule {
};
NgSqrtPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSqrtPipeModule });
NgSqrtPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSqrtPipeModule_Factory(t) { return new (t || NgSqrtPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SqrtPipe, [{
        type: Pipe,
        args: [{
                name: 'sqrt'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSqrtPipeModule, { declarations: [SqrtPipe], exports: [SqrtPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSqrtPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SqrtPipe],
                exports: [SqrtPipe]
            }]
    }], null, null); })();
export { NgSqrtPipeModule };
