// angular-pipes/esm5/string/capitalize.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, upperFirst } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var CapitalizePipe = /** @class */ (function () {
    function CapitalizePipe() {
    }
    CapitalizePipe.prototype.transform = function (input, all) {
        if (all === void 0) { all = false; }
        if (!isString(input)) {
            return input;
        }
        if (!all) {
            return upperFirst(input.toLowerCase());
        }
        else {
            return input
                .toLowerCase()
                .split(' ')
                .map(function (value) { return upperFirst(value); })
                .join(' ');
        }
    };
CapitalizePipe.ɵfac = function CapitalizePipe_Factory(t) { return new (t || CapitalizePipe)(); };
CapitalizePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "capitalize", type: CapitalizePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CapitalizePipe, [{
        type: Pipe,
        args: [{
                name: 'capitalize'
            }]
    }], function () { return []; }, null); })();
    return CapitalizePipe;
}());
export { CapitalizePipe };
var NgCapitalizePipeModule = /** @class */ (function () {
    function NgCapitalizePipeModule() {
    }
NgCapitalizePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgCapitalizePipeModule });
NgCapitalizePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgCapitalizePipeModule_Factory(t) { return new (t || NgCapitalizePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgCapitalizePipeModule, { declarations: [CapitalizePipe], exports: [CapitalizePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgCapitalizePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [CapitalizePipe],
                exports: [CapitalizePipe]
            }]
    }], function () { return []; }, null); })();
    return NgCapitalizePipeModule;
}());
export { NgCapitalizePipeModule };
