// angular-pipes/esm5/string/decode-uri-component.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var DecodeURIComponentPipe = /** @class */ (function () {
    function DecodeURIComponentPipe() {
    }
    DecodeURIComponentPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return decodeURIComponent(input);
    };
DecodeURIComponentPipe.ɵfac = function DecodeURIComponentPipe_Factory(t) { return new (t || DecodeURIComponentPipe)(); };
DecodeURIComponentPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "decodeURIComponent", type: DecodeURIComponentPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DecodeURIComponentPipe, [{
        type: Pipe,
        args: [{
                name: 'decodeURIComponent'
            }]
    }], function () { return []; }, null); })();
    return DecodeURIComponentPipe;
}());
export { DecodeURIComponentPipe };
var NgDecodeURIComponentPipeModule = /** @class */ (function () {
    function NgDecodeURIComponentPipeModule() {
    }
NgDecodeURIComponentPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDecodeURIComponentPipeModule });
NgDecodeURIComponentPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDecodeURIComponentPipeModule_Factory(t) { return new (t || NgDecodeURIComponentPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDecodeURIComponentPipeModule, { declarations: [DecodeURIComponentPipe], exports: [DecodeURIComponentPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDecodeURIComponentPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DecodeURIComponentPipe],
                exports: [DecodeURIComponentPipe]
            }]
    }], function () { return []; }, null); })();
    return NgDecodeURIComponentPipeModule;
}());
export { NgDecodeURIComponentPipeModule };
