// angular-pipes/esm5/string/decode-uri.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var DecodeURIPipe = /** @class */ (function () {
    function DecodeURIPipe() {
    }
    DecodeURIPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return decodeURI(input);
    };
DecodeURIPipe.ɵfac = function DecodeURIPipe_Factory(t) { return new (t || DecodeURIPipe)(); };
DecodeURIPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "decodeURI", type: DecodeURIPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DecodeURIPipe, [{
        type: Pipe,
        args: [{
                name: 'decodeURI'
            }]
    }], function () { return []; }, null); })();
    return DecodeURIPipe;
}());
export { DecodeURIPipe };
var NgDecodeURIPipeModule = /** @class */ (function () {
    function NgDecodeURIPipeModule() {
    }
NgDecodeURIPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDecodeURIPipeModule });
NgDecodeURIPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDecodeURIPipeModule_Factory(t) { return new (t || NgDecodeURIPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDecodeURIPipeModule, { declarations: [DecodeURIPipe], exports: [DecodeURIPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDecodeURIPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DecodeURIPipe],
                exports: [DecodeURIPipe]
            }]
    }], function () { return []; }, null); })();
    return NgDecodeURIPipeModule;
}());
export { NgDecodeURIPipeModule };
