// angular-pipes/esm5/string/encode-uri.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var EncodeURIPipe = /** @class */ (function () {
    function EncodeURIPipe() {
    }
    EncodeURIPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return encodeURI(input);
    };
EncodeURIPipe.ɵfac = function EncodeURIPipe_Factory(t) { return new (t || EncodeURIPipe)(); };
EncodeURIPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "encodeURI", type: EncodeURIPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EncodeURIPipe, [{
        type: Pipe,
        args: [{
                name: 'encodeURI'
            }]
    }], function () { return []; }, null); })();
    return EncodeURIPipe;
}());
export { EncodeURIPipe };
var NgEncodeURIPipeModule = /** @class */ (function () {
    function NgEncodeURIPipeModule() {
    }
NgEncodeURIPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEncodeURIPipeModule });
NgEncodeURIPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEncodeURIPipeModule_Factory(t) { return new (t || NgEncodeURIPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEncodeURIPipeModule, { declarations: [EncodeURIPipe], exports: [EncodeURIPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEncodeURIPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EncodeURIPipe],
                exports: [EncodeURIPipe]
            }]
    }], function () { return []; }, null); })();
    return NgEncodeURIPipeModule;
}());
export { NgEncodeURIPipeModule };
