// angular-pipes/esm5/string/encode-uri-component.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var EncodeURIComponentPipe = /** @class */ (function () {
    function EncodeURIComponentPipe() {
    }
    EncodeURIComponentPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return encodeURIComponent(input);
    };
EncodeURIComponentPipe.ɵfac = function EncodeURIComponentPipe_Factory(t) { return new (t || EncodeURIComponentPipe)(); };
EncodeURIComponentPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "encodeURIComponent", type: EncodeURIComponentPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EncodeURIComponentPipe, [{
        type: Pipe,
        args: [{
                name: 'encodeURIComponent'
            }]
    }], function () { return []; }, null); })();
    return EncodeURIComponentPipe;
}());
export { EncodeURIComponentPipe };
var NgEncodeURIComponentPipeModule = /** @class */ (function () {
    function NgEncodeURIComponentPipeModule() {
    }
NgEncodeURIComponentPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEncodeURIComponentPipeModule });
NgEncodeURIComponentPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEncodeURIComponentPipeModule_Factory(t) { return new (t || NgEncodeURIComponentPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEncodeURIComponentPipeModule, { declarations: [EncodeURIComponentPipe], exports: [EncodeURIComponentPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEncodeURIComponentPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EncodeURIComponentPipe],
                exports: [EncodeURIComponentPipe]
            }]
    }], function () { return []; }, null); })();
    return NgEncodeURIComponentPipeModule;
}());
export { NgEncodeURIComponentPipeModule };
