// angular-pipes/esm5/string/left-pad.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { leftPad, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var LeftPadPipe = /** @class */ (function () {
    function LeftPadPipe() {
    }
    LeftPadPipe.prototype.transform = function (input, length, character) {
        if (length === void 0) { length = 0; }
        if (character === void 0) { character = ' '; }
        if (!isString(input)) {
            return input;
        }
        return leftPad(input, length, character);
    };
LeftPadPipe.ɵfac = function LeftPadPipe_Factory(t) { return new (t || LeftPadPipe)(); };
LeftPadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "leftpad", type: LeftPadPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LeftPadPipe, [{
        type: Pipe,
        args: [{
                name: 'leftpad'
            }]
    }], function () { return []; }, null); })();
    return LeftPadPipe;
}());
export { LeftPadPipe };
var NgLeftPadPipeModule = /** @class */ (function () {
    function NgLeftPadPipeModule() {
    }
NgLeftPadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgLeftPadPipeModule });
NgLeftPadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgLeftPadPipeModule_Factory(t) { return new (t || NgLeftPadPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgLeftPadPipeModule, { declarations: [LeftPadPipe], exports: [LeftPadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgLeftPadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [LeftPadPipe],
                exports: [LeftPadPipe]
            }]
    }], function () { return []; }, null); })();
    return NgLeftPadPipeModule;
}());
export { NgLeftPadPipeModule };
