// angular-pipes/esm5/string/match.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var MatchPipe = /** @class */ (function () {
    function MatchPipe() {
    }
    MatchPipe.prototype.transform = function (input, pattern, flag) {
        if (!isString(input)) {
            return input;
        }
        var regexp = pattern instanceof RegExp ? pattern : new RegExp(pattern, flag);
        return input.match(regexp);
    };
MatchPipe.ɵfac = function MatchPipe_Factory(t) { return new (t || MatchPipe)(); };
MatchPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "match", type: MatchPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MatchPipe, [{
        type: Pipe,
        args: [{
                name: 'match'
            }]
    }], function () { return []; }, null); })();
    return MatchPipe;
}());
export { MatchPipe };
var NgMatchPipeModule = /** @class */ (function () {
    function NgMatchPipeModule() {
    }
NgMatchPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMatchPipeModule });
NgMatchPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMatchPipeModule_Factory(t) { return new (t || NgMatchPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMatchPipeModule, { declarations: [MatchPipe], exports: [MatchPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMatchPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MatchPipe],
                exports: [MatchPipe]
            }]
    }], function () { return []; }, null); })();
    return NgMatchPipeModule;
}());
export { NgMatchPipeModule };
