// angular-pipes/esm5/string/repeat.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var RepeatPipe = /** @class */ (function () {
    function RepeatPipe() {
    }
    RepeatPipe.prototype.transform = function (input, times, characters) {
        if (times === void 0) { times = 1; }
        if (characters === void 0) { characters = ''; }
        if (!isString(input)) {
            return input;
        }
        if (times <= 0) {
            times = 1;
        }
        var repeated = [input];
        for (var i = 1; i < times; ++i) {
            repeated.push(input);
        }
        return repeated.join(characters);
    };
RepeatPipe.ɵfac = function RepeatPipe_Factory(t) { return new (t || RepeatPipe)(); };
RepeatPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "repeat", type: RepeatPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RepeatPipe, [{
        type: Pipe,
        args: [{
                name: 'repeat'
            }]
    }], function () { return []; }, null); })();
    return RepeatPipe;
}());
export { RepeatPipe };
var NgRepeatPipeModule = /** @class */ (function () {
    function NgRepeatPipeModule() {
    }
NgRepeatPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRepeatPipeModule });
NgRepeatPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRepeatPipeModule_Factory(t) { return new (t || NgRepeatPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRepeatPipeModule, { declarations: [RepeatPipe], exports: [RepeatPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRepeatPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RepeatPipe],
                exports: [RepeatPipe]
            }]
    }], function () { return []; }, null); })();
    return NgRepeatPipeModule;
}());
export { NgRepeatPipeModule };
