// angular-pipes/esm5/string/replace.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var ReplacePipe = /** @class */ (function () {
    function ReplacePipe() {
    }
    ReplacePipe.prototype.transform = function (input, pattern, replacement) {
        if (!isString(input) || isUndefined(pattern) || isUndefined(replacement)) {
            return input;
        }
        return input.replace(pattern, replacement);
    };
ReplacePipe.ɵfac = function ReplacePipe_Factory(t) { return new (t || ReplacePipe)(); };
ReplacePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "replace", type: ReplacePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ReplacePipe, [{
        type: Pipe,
        args: [{
                name: 'replace'
            }]
    }], function () { return []; }, null); })();
    return ReplacePipe;
}());
export { ReplacePipe };
var NgReplacePipeModule = /** @class */ (function () {
    function NgReplacePipeModule() {
    }
NgReplacePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgReplacePipeModule });
NgReplacePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgReplacePipeModule_Factory(t) { return new (t || NgReplacePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgReplacePipeModule, { declarations: [ReplacePipe], exports: [ReplacePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgReplacePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ReplacePipe],
                exports: [ReplacePipe]
            }]
    }], function () { return []; }, null); })();
    return NgReplacePipeModule;
}());
export { NgReplacePipeModule };
