// angular-pipes/esm5/string/right-pad.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { rightPad, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var RightPadPipe = /** @class */ (function () {
    function RightPadPipe() {
    }
    RightPadPipe.prototype.transform = function (input, length, character) {
        if (length === void 0) { length = 0; }
        if (character === void 0) { character = ' '; }
        if (!isString(input)) {
            return input;
        }
        return rightPad(input, length, character);
    };
RightPadPipe.ɵfac = function RightPadPipe_Factory(t) { return new (t || RightPadPipe)(); };
RightPadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "rightpad", type: RightPadPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RightPadPipe, [{
        type: Pipe,
        args: [{
                name: 'rightpad'
            }]
    }], function () { return []; }, null); })();
    return RightPadPipe;
}());
export { RightPadPipe };
var NgRightPadPipeModule = /** @class */ (function () {
    function NgRightPadPipeModule() {
    }
NgRightPadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRightPadPipeModule });
NgRightPadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRightPadPipeModule_Factory(t) { return new (t || NgRightPadPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRightPadPipeModule, { declarations: [RightPadPipe], exports: [RightPadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRightPadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RightPadPipe],
                exports: [RightPadPipe]
            }]
    }], function () { return []; }, null); })();
    return NgRightPadPipeModule;
}());
export { NgRightPadPipeModule };
