// angular-pipes/esm5/string/split.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var SplitPipe = /** @class */ (function () {
    function SplitPipe() {
    }
    SplitPipe.prototype.transform = function (input, separator, limit) {
        if (separator === void 0) { separator = ' '; }
        if (!isString(input)) {
            return input;
        }
        return input.split(separator, limit);
    };
SplitPipe.ɵfac = function SplitPipe_Factory(t) { return new (t || SplitPipe)(); };
SplitPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "split", type: SplitPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SplitPipe, [{
        type: Pipe,
        args: [{
                name: 'split'
            }]
    }], function () { return []; }, null); })();
    return SplitPipe;
}());
export { SplitPipe };
var NgSplitPipeModule = /** @class */ (function () {
    function NgSplitPipeModule() {
    }
NgSplitPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSplitPipeModule });
NgSplitPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSplitPipeModule_Factory(t) { return new (t || NgSplitPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSplitPipeModule, { declarations: [SplitPipe], exports: [SplitPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSplitPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SplitPipe],
                exports: [SplitPipe]
            }]
    }], function () { return []; }, null); })();
    return NgSplitPipeModule;
}());
export { NgSplitPipeModule };
