// angular-pipes/esm5/string/strip-tags.pipe.js
// idea from https://github.com/a8m/angular-filter
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var StripTagsPipe = /** @class */ (function () {
    function StripTagsPipe() {
    }
    StripTagsPipe.prototype.transform = function (input) {
        if (!isString(input) || isUndefined(input))
            return input;
        return input.replace(/<\S[^><]*>/g, '');
    };
StripTagsPipe.ɵfac = function StripTagsPipe_Factory(t) { return new (t || StripTagsPipe)(); };
StripTagsPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "stripTags", type: StripTagsPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StripTagsPipe, [{
        type: Pipe,
        args: [{
                name: 'stripTags'
            }]
    }], function () { return []; }, null); })();
    return StripTagsPipe;
}());
export { StripTagsPipe };
var NgStripTagsPipeModule = /** @class */ (function () {
    function NgStripTagsPipeModule() {
    }
NgStripTagsPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgStripTagsPipeModule });
NgStripTagsPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgStripTagsPipeModule_Factory(t) { return new (t || NgStripTagsPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgStripTagsPipeModule, { declarations: [StripTagsPipe], exports: [StripTagsPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgStripTagsPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [StripTagsPipe],
                exports: [StripTagsPipe]
            }]
    }], function () { return []; }, null); })();
    return NgStripTagsPipeModule;
}());
export { NgStripTagsPipeModule };
