// angular-pipes/esm5/string/test.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var TestPipe = /** @class */ (function () {
    function TestPipe() {
    }
    TestPipe.prototype.transform = function (input, pattern, flag) {
        if (!isString(input) || !pattern) {
            return input;
        }
        var regexp = pattern instanceof RegExp ? pattern : new RegExp(pattern, flag);
        return regexp.test(input);
    };
TestPipe.ɵfac = function TestPipe_Factory(t) { return new (t || TestPipe)(); };
TestPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "test", type: TestPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TestPipe, [{
        type: Pipe,
        args: [{
                name: 'test'
            }]
    }], function () { return []; }, null); })();
    return TestPipe;
}());
export { TestPipe };
var NgTestPipeModule = /** @class */ (function () {
    function NgTestPipeModule() {
    }
NgTestPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTestPipeModule });
NgTestPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTestPipeModule_Factory(t) { return new (t || NgTestPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTestPipeModule, { declarations: [TestPipe], exports: [TestPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTestPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TestPipe],
                exports: [TestPipe]
            }]
    }], function () { return []; }, null); })();
    return NgTestPipeModule;
}());
export { NgTestPipeModule };
