// angular-pipes/esm5/string/trim.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var TrimPipe = /** @class */ (function () {
    function TrimPipe() {
    }
    TrimPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return input.trim();
    };
TrimPipe.ɵfac = function TrimPipe_Factory(t) { return new (t || TrimPipe)(); };
TrimPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "trim", type: TrimPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TrimPipe, [{
        type: Pipe,
        args: [{
                name: 'trim'
            }]
    }], function () { return []; }, null); })();
    return TrimPipe;
}());
export { TrimPipe };
var NgTrimPipeModule = /** @class */ (function () {
    function NgTrimPipeModule() {
    }
NgTrimPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTrimPipeModule });
NgTrimPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTrimPipeModule_Factory(t) { return new (t || NgTrimPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTrimPipeModule, { declarations: [TrimPipe], exports: [TrimPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTrimPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TrimPipe],
                exports: [TrimPipe]
            }]
    }], function () { return []; }, null); })();
    return NgTrimPipeModule;
}());
export { NgTrimPipeModule };
