// angular-pipes/esm5/string/upperfirst.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, upperFirst } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var UpperFirstPipe = /** @class */ (function () {
    function UpperFirstPipe() {
    }
    UpperFirstPipe.prototype.transform = function (input) {
        if (!isString(input)) {
            return input;
        }
        return upperFirst(input);
    };
UpperFirstPipe.ɵfac = function UpperFirstPipe_Factory(t) { return new (t || UpperFirstPipe)(); };
UpperFirstPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "upperfirst", type: UpperFirstPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UpperFirstPipe, [{
        type: Pipe,
        args: [{
                name: 'upperfirst'
            }]
    }], function () { return []; }, null); })();
    return UpperFirstPipe;
}());
export { UpperFirstPipe };
var NgUpperFirstPipeModule = /** @class */ (function () {
    function NgUpperFirstPipeModule() {
    }
NgUpperFirstPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUpperFirstPipeModule });
NgUpperFirstPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUpperFirstPipeModule_Factory(t) { return new (t || NgUpperFirstPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUpperFirstPipeModule, { declarations: [UpperFirstPipe], exports: [UpperFirstPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUpperFirstPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [UpperFirstPipe],
                exports: [UpperFirstPipe]
            }]
    }], function () { return []; }, null); })();
    return NgUpperFirstPipeModule;
}());
export { NgUpperFirstPipeModule };
