// angular-pipes/esm5/string/with.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isNull } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var WithPipe = /** @class */ (function () {
    function WithPipe() {
    }
    WithPipe.prototype.transform = function (input, start, ends, csensitive) {
        if (start === void 0) { start = null; }
        if (ends === void 0) { ends = null; }
        if (csensitive === void 0) { csensitive = false; }
        if (!isString(input) || (isNull(start) && isNull(ends)) || start == '' || ends == '') {
            return input;
        }
        input = csensitive ? input : input.toLowerCase();
        if (!isNull(start) && !isNull(ends)) {
            var a = !input.indexOf(csensitive ? start : start.toLowerCase());
            var b = input.indexOf(csensitive ? ends : ends.toLowerCase(), input.length - ends.length) !== -1;
            if (a == true && b == true) {
                return true;
            }
            else {
                return false;
            }
        }
        if (!isNull(start)) {
            return !input.indexOf(csensitive ? start : start.toLowerCase());
        }
        if (!isNull(ends)) {
            var position = input.length - ends.length;
            return input.indexOf(csensitive ? ends : ends.toLowerCase(), position) !== -1;
        }
    };
WithPipe.ɵfac = function WithPipe_Factory(t) { return new (t || WithPipe)(); };
WithPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "with", type: WithPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WithPipe, [{
        type: Pipe,
        args: [{ name: 'with' }]
    }], function () { return []; }, null); })();
    return WithPipe;
}());
export { WithPipe };
var NgWithPipeModule = /** @class */ (function () {
    function NgWithPipeModule() {
    }
NgWithPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgWithPipeModule });
NgWithPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgWithPipeModule_Factory(t) { return new (t || NgWithPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgWithPipeModule, { declarations: [WithPipe], exports: [WithPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgWithPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [WithPipe],
                exports: [WithPipe]
            }]
    }], function () { return []; }, null); })();
    return NgWithPipeModule;
}());
export { NgWithPipeModule };
