// angular-pipes/esm2015/aggregate/group-by.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { getProperty, isArray, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let GroupByPipe = class GroupByPipe {
    transform(input, prop) {
        if (!isArray(input)) {
            return input;
        }
        const arr = {};
        for (const value of input) {
            const field = getProperty(value, prop);
            if (isUndefined(arr[field])) {
                arr[field] = [];
            }
            arr[field].push(value);
        }
        return Object.keys(arr).map(key => ({ key, value: arr[key] }));
    }
};
GroupByPipe.ɵfac = function GroupByPipe_Factory(t) { return new (t || GroupByPipe)(); };
GroupByPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "groupBy", type: GroupByPipe, pure: true });
export { GroupByPipe };
let NgGroupByPipeModule = class NgGroupByPipeModule {
};
NgGroupByPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgGroupByPipeModule });
NgGroupByPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgGroupByPipeModule_Factory(t) { return new (t || NgGroupByPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GroupByPipe, [{
        type: Pipe,
        args: [{
                name: 'groupBy'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgGroupByPipeModule, { declarations: [GroupByPipe], exports: [GroupByPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgGroupByPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [GroupByPipe],
                exports: [GroupByPipe]
            }]
    }], null, null); })();
export { NgGroupByPipeModule };
