// angular-pipes/esm2015/aggregate/max.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let MaxPipe = class MaxPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        if (input.length === 0) {
            return undefined;
        }
        let max = input[0];
        input.forEach((value) => {
            if (max < value) {
                max = value;
            }
        });
        return max;
    }
};
MaxPipe.ɵfac = function MaxPipe_Factory(t) { return new (t || MaxPipe)(); };
MaxPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "max", type: MaxPipe, pure: true });
export { MaxPipe };
let NgMaxPipeModule = class NgMaxPipeModule {
};
NgMaxPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMaxPipeModule });
NgMaxPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMaxPipeModule_Factory(t) { return new (t || NgMaxPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaxPipe, [{
        type: Pipe,
        args: [{
                name: 'max'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMaxPipeModule, { declarations: [MaxPipe], exports: [MaxPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMaxPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MaxPipe],
                exports: [MaxPipe]
            }]
    }], null, null); })();
export { NgMaxPipeModule };
