// angular-pipes/esm2015/aggregate/mean.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, sum } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let MeanPipe = class MeanPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        const count = input.length;
        if (count === 0) {
            return undefined;
        }
        return sum(input) / count;
    }
};
MeanPipe.ɵfac = function MeanPipe_Factory(t) { return new (t || MeanPipe)(); };
MeanPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "mean", type: MeanPipe, pure: true });
export { MeanPipe };
let NgMeanPipeModule = class NgMeanPipeModule {
};
NgMeanPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMeanPipeModule });
NgMeanPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMeanPipeModule_Factory(t) { return new (t || NgMeanPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MeanPipe, [{
        type: Pipe,
        args: [{
                name: 'mean'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMeanPipeModule, { declarations: [MeanPipe], exports: [MeanPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMeanPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MeanPipe],
                exports: [MeanPipe]
            }]
    }], null, null); })();
export { NgMeanPipeModule };
