// angular-pipes/esm2015/aggregate/min.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let MinPipe = class MinPipe {
    transform(input) {
        if (!isArray(input)) {
            return input;
        }
        if (input.length === 0) {
            return undefined;
        }
        let min = input[0];
        input.forEach((value) => {
            if (min > value) {
                min = value;
            }
        });
        return min;
    }
};
MinPipe.ɵfac = function MinPipe_Factory(t) { return new (t || MinPipe)(); };
MinPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "min", type: MinPipe, pure: true });
export { MinPipe };
let NgMinPipeModule = class NgMinPipeModule {
};
NgMinPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMinPipeModule });
NgMinPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMinPipeModule_Factory(t) { return new (t || NgMinPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MinPipe, [{
        type: Pipe,
        args: [{
                name: 'min'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMinPipeModule, { declarations: [MinPipe], exports: [MinPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMinPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MinPipe],
                exports: [MinPipe]
            }]
    }], null, null); })();
export { NgMinPipeModule };
