// angular-pipes/esm2015/boolean/is-array.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IsArrayPipe = class IsArrayPipe {
    transform(value) {
        return isArray(value);
    }
};
IsArrayPipe.ɵfac = function IsArrayPipe_Factory(t) { return new (t || IsArrayPipe)(); };
IsArrayPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isArray", type: IsArrayPipe, pure: true });
export { IsArrayPipe };
let NgIsArrayPipeModule = class NgIsArrayPipeModule {
};
NgIsArrayPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsArrayPipeModule });
NgIsArrayPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsArrayPipeModule_Factory(t) { return new (t || NgIsArrayPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsArrayPipe, [{
        type: Pipe,
        args: [{
                name: 'isArray'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsArrayPipeModule, { declarations: [IsArrayPipe], exports: [IsArrayPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsArrayPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsArrayPipe],
                exports: [IsArrayPipe]
            }]
    }], null, null); })();
export { NgIsArrayPipeModule };
