// angular-pipes/esm2015/boolean/is-defined.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IsDefinedPipe = class IsDefinedPipe {
    transform(value) {
        return !isUndefined(value);
    }
};
IsDefinedPipe.ɵfac = function IsDefinedPipe_Factory(t) { return new (t || IsDefinedPipe)(); };
IsDefinedPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isDefined", type: IsDefinedPipe, pure: true });
export { IsDefinedPipe };
let NgIsDefinedPipeModule = class NgIsDefinedPipeModule {
};
NgIsDefinedPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsDefinedPipeModule });
NgIsDefinedPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsDefinedPipeModule_Factory(t) { return new (t || NgIsDefinedPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsDefinedPipe, [{
        type: Pipe,
        args: [{
                name: 'isDefined'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsDefinedPipeModule, { declarations: [IsDefinedPipe], exports: [IsDefinedPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsDefinedPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsDefinedPipe],
                exports: [IsDefinedPipe]
            }]
    }], null, null); })();
export { NgIsDefinedPipeModule };
