// angular-pipes/esm2015/boolean/is-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsEqualPipe = class IsEqualPipe {
    transform(first, second) {
        return first == second;
    }
};
IsEqualPipe.ɵfac = function IsEqualPipe_Factory(t) { return new (t || IsEqualPipe)(); };
IsEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "equal", type: IsEqualPipe, pure: true });
export { IsEqualPipe };
let NgIsEqualPipeModule = class NgIsEqualPipeModule {
};
NgIsEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsEqualPipeModule });
NgIsEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsEqualPipeModule_Factory(t) { return new (t || NgIsEqualPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'equal'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsEqualPipeModule, { declarations: [IsEqualPipe], exports: [IsEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsEqualPipe],
                exports: [IsEqualPipe]
            }]
    }], null, null); })();
export { NgIsEqualPipeModule };
