// angular-pipes/esm2015/boolean/is-greater-or-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsGreaterOrEqualPipe = class IsGreaterOrEqualPipe {
    transform(first, second) {
        return first >= second;
    }
};
IsGreaterOrEqualPipe.ɵfac = function IsGreaterOrEqualPipe_Factory(t) { return new (t || IsGreaterOrEqualPipe)(); };
IsGreaterOrEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "greaterOrEqual", type: IsGreaterOrEqualPipe, pure: true });
export { IsGreaterOrEqualPipe };
let NgIsGreaterOrEqualPipeModule = class NgIsGreaterOrEqualPipeModule {
};
NgIsGreaterOrEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsGreaterOrEqualPipeModule });
NgIsGreaterOrEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsGreaterOrEqualPipeModule_Factory(t) { return new (t || NgIsGreaterOrEqualPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsGreaterOrEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'greaterOrEqual'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsGreaterOrEqualPipeModule, { declarations: [IsGreaterOrEqualPipe], exports: [IsGreaterOrEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsGreaterOrEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsGreaterOrEqualPipe],
                exports: [IsGreaterOrEqualPipe]
            }]
    }], null, null); })();
export { NgIsGreaterOrEqualPipeModule };
