// angular-pipes/esm2015/boolean/is-identical.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsIdenticalPipe = class IsIdenticalPipe {
    transform(first, second) {
        return first === second;
    }
};
IsIdenticalPipe.ɵfac = function IsIdenticalPipe_Factory(t) { return new (t || IsIdenticalPipe)(); };
IsIdenticalPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "identical", type: IsIdenticalPipe, pure: true });
export { IsIdenticalPipe };
let NgIsIdenticalPipeModule = class NgIsIdenticalPipeModule {
};
NgIsIdenticalPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsIdenticalPipeModule });
NgIsIdenticalPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsIdenticalPipeModule_Factory(t) { return new (t || NgIsIdenticalPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsIdenticalPipe, [{
        type: Pipe,
        args: [{
                name: 'identical'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsIdenticalPipeModule, { declarations: [IsIdenticalPipe], exports: [IsIdenticalPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsIdenticalPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsIdenticalPipe],
                exports: [IsIdenticalPipe]
            }]
    }], null, null); })();
export { NgIsIdenticalPipeModule };
