// angular-pipes/esm2015/boolean/is-less.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsLessPipe = class IsLessPipe {
    transform(first, second) {
        return first < second;
    }
};
IsLessPipe.ɵfac = function IsLessPipe_Factory(t) { return new (t || IsLessPipe)(); };
IsLessPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "less", type: IsLessPipe, pure: true });
export { IsLessPipe };
let NgIsLessPipeModule = class NgIsLessPipeModule {
};
NgIsLessPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsLessPipeModule });
NgIsLessPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsLessPipeModule_Factory(t) { return new (t || NgIsLessPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsLessPipe, [{
        type: Pipe,
        args: [{
                name: 'less'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsLessPipeModule, { declarations: [IsLessPipe], exports: [IsLessPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsLessPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsLessPipe],
                exports: [IsLessPipe]
            }]
    }], null, null); })();
export { NgIsLessPipeModule };
