// angular-pipes/esm2015/boolean/is-less-or-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsLessOrEqualPipe = class IsLessOrEqualPipe {
    transform(first, second) {
        return first <= second;
    }
};
IsLessOrEqualPipe.ɵfac = function IsLessOrEqualPipe_Factory(t) { return new (t || IsLessOrEqualPipe)(); };
IsLessOrEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "lessOrEqual", type: IsLessOrEqualPipe, pure: true });
export { IsLessOrEqualPipe };
let NgIsLessOrEqualPipeModule = class NgIsLessOrEqualPipeModule {
};
NgIsLessOrEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsLessOrEqualPipeModule });
NgIsLessOrEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsLessOrEqualPipeModule_Factory(t) { return new (t || NgIsLessOrEqualPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsLessOrEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'lessOrEqual'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsLessOrEqualPipeModule, { declarations: [IsLessOrEqualPipe], exports: [IsLessOrEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsLessOrEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsLessOrEqualPipe],
                exports: [IsLessOrEqualPipe]
            }]
    }], null, null); })();
export { NgIsLessOrEqualPipeModule };
