// angular-pipes/esm2015/boolean/is-not-equal.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsNotEqualPipe = class IsNotEqualPipe {
    transform(first, second) {
        return first != second;
    }
};
IsNotEqualPipe.ɵfac = function IsNotEqualPipe_Factory(t) { return new (t || IsNotEqualPipe)(); };
IsNotEqualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "notEqual", type: IsNotEqualPipe, pure: true });
export { IsNotEqualPipe };
let NgIsNotEqualPipeModule = class NgIsNotEqualPipeModule {
};
NgIsNotEqualPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNotEqualPipeModule });
NgIsNotEqualPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNotEqualPipeModule_Factory(t) { return new (t || NgIsNotEqualPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNotEqualPipe, [{
        type: Pipe,
        args: [{
                name: 'notEqual'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNotEqualPipeModule, { declarations: [IsNotEqualPipe], exports: [IsNotEqualPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNotEqualPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNotEqualPipe],
                exports: [IsNotEqualPipe]
            }]
    }], null, null); })();
export { NgIsNotEqualPipeModule };
