// angular-pipes/esm2015/boolean/is-null.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNull } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IsNullPipe = class IsNullPipe {
    transform(value) {
        return isNull(value);
    }
};
IsNullPipe.ɵfac = function IsNullPipe_Factory(t) { return new (t || IsNullPipe)(); };
IsNullPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isNull", type: IsNullPipe, pure: true });
export { IsNullPipe };
let NgIsNullPipeModule = class NgIsNullPipeModule {
};
NgIsNullPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNullPipeModule });
NgIsNullPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNullPipeModule_Factory(t) { return new (t || NgIsNullPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNullPipe, [{
        type: Pipe,
        args: [{
                name: 'isNull'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNullPipeModule, { declarations: [IsNullPipe], exports: [IsNullPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNullPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNullPipe],
                exports: [IsNullPipe]
            }]
    }], null, null); })();
export { NgIsNullPipeModule };
