// angular-pipes/esm2015/boolean/is-number.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumber } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IsNumberPipe = class IsNumberPipe {
    transform(value) {
        return isNumber(value);
    }
};
IsNumberPipe.ɵfac = function IsNumberPipe_Factory(t) { return new (t || IsNumberPipe)(); };
IsNumberPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isNumber", type: IsNumberPipe, pure: true });
export { IsNumberPipe };
let NgIsNumberPipeModule = class NgIsNumberPipeModule {
};
NgIsNumberPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNumberPipeModule });
NgIsNumberPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNumberPipeModule_Factory(t) { return new (t || NgIsNumberPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNumberPipe, [{
        type: Pipe,
        args: [{
                name: 'isNumber'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNumberPipeModule, { declarations: [IsNumberPipe], exports: [IsNumberPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNumberPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNumberPipe],
                exports: [IsNumberPipe]
            }]
    }], null, null); })();
export { NgIsNumberPipeModule };
