// angular-pipes/esm2015/boolean/is-not-identical.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let IsNotIdenticalPipe = class IsNotIdenticalPipe {
    transform(first, second) {
        return first !== second;
    }
};
IsNotIdenticalPipe.ɵfac = function IsNotIdenticalPipe_Factory(t) { return new (t || IsNotIdenticalPipe)(); };
IsNotIdenticalPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "notIdentical", type: IsNotIdenticalPipe, pure: true });
export { IsNotIdenticalPipe };
let NgIsNotIdenticalPipeModule = class NgIsNotIdenticalPipeModule {
};
NgIsNotIdenticalPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNotIdenticalPipeModule });
NgIsNotIdenticalPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNotIdenticalPipeModule_Factory(t) { return new (t || NgIsNotIdenticalPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNotIdenticalPipe, [{
        type: Pipe,
        args: [{
                name: 'notIdentical'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNotIdenticalPipeModule, { declarations: [IsNotIdenticalPipe], exports: [IsNotIdenticalPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNotIdenticalPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNotIdenticalPipe],
                exports: [IsNotIdenticalPipe]
            }]
    }], null, null); })();
export { NgIsNotIdenticalPipeModule };
