// angular-pipes/esm2015/boolean/is-nil.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNil } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let IsNilPipe = class IsNilPipe {
    transform(value) {
        return isNil(value);
    }
};
IsNilPipe.ɵfac = function IsNilPipe_Factory(t) { return new (t || IsNilPipe)(); };
IsNilPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isNil", type: IsNilPipe, pure: true });
export { IsNilPipe };
let NgIsNilPipeModule = class NgIsNilPipeModule {
};
NgIsNilPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgIsNilPipeModule });
NgIsNilPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgIsNilPipeModule_Factory(t) { return new (t || NgIsNilPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IsNilPipe, [{
        type: Pipe,
        args: [{
                name: 'isNil'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgIsNilPipeModule, { declarations: [IsNilPipe], exports: [IsNilPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgIsNilPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [IsNilPipe],
                exports: [IsNilPipe]
            }]
    }], null, null); })();
export { NgIsNilPipeModule };
