// angular-pipes/esm5/math/ceil.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { createRound, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var CeilPipe = /** @class */ (function () {
    function CeilPipe() {
    }
    CeilPipe.prototype.transform = function (value, precision) {
        if (precision === void 0) { precision = 0; }
        if (isString(precision)) {
            precision = parseInt(precision);
        }
        return createRound('ceil')(value, precision);
    };
CeilPipe.ɵfac = function CeilPipe_Factory(t) { return new (t || CeilPipe)(); };
CeilPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "ceil", type: CeilPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CeilPipe, [{
        type: Pipe,
        args: [{
                name: 'ceil'
            }]
    }], function () { return []; }, null); })();
    return CeilPipe;
}());
export { CeilPipe };
var NgCeilPipeModule = /** @class */ (function () {
    function NgCeilPipeModule() {
    }
NgCeilPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgCeilPipeModule });
NgCeilPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgCeilPipeModule_Factory(t) { return new (t || NgCeilPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgCeilPipeModule, { declarations: [CeilPipe], exports: [CeilPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgCeilPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [CeilPipe],
                exports: [CeilPipe]
            }]
    }], function () { return []; }, null); })();
    return NgCeilPipeModule;
}());
export { NgCeilPipeModule };
