// angular-pipes/esm5/math/degrees.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var DegreesPipe = /** @class */ (function () {
    function DegreesPipe() {
    }
    DegreesPipe.prototype.transform = function (input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return (input * 180) / Math.PI;
    };
DegreesPipe.ɵfac = function DegreesPipe_Factory(t) { return new (t || DegreesPipe)(); };
DegreesPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "degrees", type: DegreesPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DegreesPipe, [{
        type: Pipe,
        args: [{
                name: 'degrees'
            }]
    }], function () { return []; }, null); })();
    return DegreesPipe;
}());
export { DegreesPipe };
var NgDegreesPipeModule = /** @class */ (function () {
    function NgDegreesPipeModule() {
    }
NgDegreesPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDegreesPipeModule });
NgDegreesPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDegreesPipeModule_Factory(t) { return new (t || NgDegreesPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDegreesPipeModule, { declarations: [DegreesPipe], exports: [DegreesPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDegreesPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DegreesPipe],
                exports: [DegreesPipe]
            }]
    }], function () { return []; }, null); })();
    return NgDegreesPipeModule;
}());
export { NgDegreesPipeModule };
