// angular-pipes/esm5/math/pow.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var PowPipe = /** @class */ (function () {
    function PowPipe() {
    }
    PowPipe.prototype.transform = function (input, power) {
        if (power === void 0) { power = 2; }
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return Math.pow(input, power);
    };
PowPipe.ɵfac = function PowPipe_Factory(t) { return new (t || PowPipe)(); };
PowPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "pow", type: PowPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PowPipe, [{
        type: Pipe,
        args: [{
                name: 'pow'
            }]
    }], function () { return []; }, null); })();
    return PowPipe;
}());
export { PowPipe };
var NgPowPipeModule = /** @class */ (function () {
    function NgPowPipeModule() {
    }
NgPowPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgPowPipeModule });
NgPowPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgPowPipeModule_Factory(t) { return new (t || NgPowPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgPowPipeModule, { declarations: [PowPipe], exports: [PowPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgPowPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [PowPipe],
                exports: [PowPipe]
            }]
    }], function () { return []; }, null); })();
    return NgPowPipeModule;
}());
export { NgPowPipeModule };
