// angular-pipes/esm5/math/radians.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isNumberFinite } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var RadiansPipe = /** @class */ (function () {
    function RadiansPipe() {
    }
    RadiansPipe.prototype.transform = function (input) {
        if (!isNumberFinite(input)) {
            return 'NaN';
        }
        return (input * Math.PI) / 180;
    };
RadiansPipe.ɵfac = function RadiansPipe_Factory(t) { return new (t || RadiansPipe)(); };
RadiansPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "radians", type: RadiansPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RadiansPipe, [{
        type: Pipe,
        args: [{
                name: 'radians'
            }]
    }], function () { return []; }, null); })();
    return RadiansPipe;
}());
export { RadiansPipe };
var NgRadiansPipeModule = /** @class */ (function () {
    function NgRadiansPipeModule() {
    }
NgRadiansPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRadiansPipeModule });
NgRadiansPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRadiansPipeModule_Factory(t) { return new (t || NgRadiansPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRadiansPipeModule, { declarations: [RadiansPipe], exports: [RadiansPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRadiansPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RadiansPipe],
                exports: [RadiansPipe]
            }]
    }], function () { return []; }, null); })();
    return NgRadiansPipeModule;
}());
export { NgRadiansPipeModule };
