// angular-pipes/esm5/math/round.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { createRound, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var RoundPipe = /** @class */ (function () {
    function RoundPipe() {
    }
    RoundPipe.prototype.transform = function (value, precision) {
        if (precision === void 0) { precision = 0; }
        if (isString(precision)) {
            precision = parseInt(precision);
        }
        return createRound('round')(value, precision);
    };
RoundPipe.ɵfac = function RoundPipe_Factory(t) { return new (t || RoundPipe)(); };
RoundPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "round", type: RoundPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RoundPipe, [{
        type: Pipe,
        args: [{
                name: 'round'
            }]
    }], function () { return []; }, null); })();
    return RoundPipe;
}());
export { RoundPipe };
var NgRoundPipeModule = /** @class */ (function () {
    function NgRoundPipeModule() {
    }
NgRoundPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRoundPipeModule });
NgRoundPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRoundPipeModule_Factory(t) { return new (t || NgRoundPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRoundPipeModule, { declarations: [RoundPipe], exports: [RoundPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRoundPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RoundPipe],
                exports: [RoundPipe]
            }]
    }], function () { return []; }, null); })();
    return NgRoundPipeModule;
}());
export { NgRoundPipeModule };
