// angular-pipes/esm5/object/defaults.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isObject, isArray, isNil } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var DefaultsPipe = /** @class */ (function () {
    function DefaultsPipe() {
    }
    DefaultsPipe.prototype.transform = function (input, defaults) {
        if (!isObject(defaults)) {
            return input;
        }
        if (isNil(input)) {
            return defaults;
        }
        if (isArray(input)) {
            return input.map(function (item) {
                if (isObject(item)) {
                    return Object.assign({}, defaults, item);
                }
                if (isNil(item)) {
                    return defaults;
                }
                return item;
            });
        }
        if (isObject(input)) {
            return Object.assign({}, defaults, input);
        }
        return input;
    };
DefaultsPipe.ɵfac = function DefaultsPipe_Factory(t) { return new (t || DefaultsPipe)(); };
DefaultsPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "defaults", type: DefaultsPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DefaultsPipe, [{
        type: Pipe,
        args: [{ name: 'defaults' }]
    }], function () { return []; }, null); })();
    return DefaultsPipe;
}());
export { DefaultsPipe };
var NgDefaultsPipeModule = /** @class */ (function () {
    function NgDefaultsPipeModule() {
    }
NgDefaultsPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDefaultsPipeModule });
NgDefaultsPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDefaultsPipeModule_Factory(t) { return new (t || NgDefaultsPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDefaultsPipeModule, { declarations: [DefaultsPipe], exports: [DefaultsPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDefaultsPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DefaultsPipe],
                exports: [DefaultsPipe]
            }]
    }], function () { return []; }, null); })();
    return NgDefaultsPipeModule;
}());
export { NgDefaultsPipeModule };
