// angular-pipes/esm5/object/keys.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isObject } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var KeysPipe = /** @class */ (function () {
    function KeysPipe() {
    }
    KeysPipe.prototype.transform = function (input) {
        if (!isObject(input)) {
            return input;
        }
        return Object.keys(input);
    };
KeysPipe.ɵfac = function KeysPipe_Factory(t) { return new (t || KeysPipe)(); };
KeysPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "keys", type: KeysPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(KeysPipe, [{
        type: Pipe,
        args: [{ name: 'keys' }]
    }], function () { return []; }, null); })();
    return KeysPipe;
}());
export { KeysPipe };
var NgKeysPipeModule = /** @class */ (function () {
    function NgKeysPipeModule() {
    }
NgKeysPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgKeysPipeModule });
NgKeysPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgKeysPipeModule_Factory(t) { return new (t || NgKeysPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgKeysPipeModule, { declarations: [KeysPipe], exports: [KeysPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgKeysPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [KeysPipe],
                exports: [KeysPipe]
            }]
    }], function () { return []; }, null); })();
    return NgKeysPipeModule;
}());
export { NgKeysPipeModule };
