// angular-pipes/esm5/object/to-array.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isObject } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var ToArrayPipe = /** @class */ (function () {
    function ToArrayPipe() {
    }
    ToArrayPipe.prototype.transform = function (input) {
        if (!isObject(input)) {
            return input;
        }
        return Object.keys(input).map(function (value) { return input[value]; });
    };
ToArrayPipe.ɵfac = function ToArrayPipe_Factory(t) { return new (t || ToArrayPipe)(); };
ToArrayPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "toArray", type: ToArrayPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ToArrayPipe, [{
        type: Pipe,
        args: [{ name: 'toArray' }]
    }], function () { return []; }, null); })();
    return ToArrayPipe;
}());
export { ToArrayPipe };
var NgToArrayPipeModule = /** @class */ (function () {
    function NgToArrayPipeModule() {
    }
NgToArrayPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgToArrayPipeModule });
NgToArrayPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgToArrayPipeModule_Factory(t) { return new (t || NgToArrayPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgToArrayPipeModule, { declarations: [ToArrayPipe], exports: [ToArrayPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgToArrayPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ToArrayPipe],
                exports: [ToArrayPipe]
            }]
    }], function () { return []; }, null); })();
    return NgToArrayPipeModule;
}());
export { NgToArrayPipeModule };
