// angular-pipes/esm5/array/chunk.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var ChunkPipe = /** @class */ (function () {
    function ChunkPipe() {
    }
    ChunkPipe.prototype.transform = function (input, size) {
        if (size === void 0) { size = 1; }
        if (!isArray(input)) {
            return input;
        }
        return [].concat.apply([], input.map(function (_elem, i) {
            return i % size ? [] : [input.slice(i, i + size)];
        }));
    };
ChunkPipe.ɵfac = function ChunkPipe_Factory(t) { return new (t || ChunkPipe)(); };
ChunkPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "chunk", type: ChunkPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChunkPipe, [{
        type: Pipe,
        args: [{
                name: 'chunk'
            }]
    }], function () { return []; }, null); })();
    return ChunkPipe;
}());
export { ChunkPipe };
var NgChunkPipeModule = /** @class */ (function () {
    function NgChunkPipeModule() {
    }
NgChunkPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgChunkPipeModule });
NgChunkPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgChunkPipeModule_Factory(t) { return new (t || NgChunkPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgChunkPipeModule, { declarations: [ChunkPipe], exports: [ChunkPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgChunkPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ChunkPipe],
                exports: [ChunkPipe]
            }]
    }], function () { return []; }, null); })();
    return NgChunkPipeModule;
}());
export { NgChunkPipeModule };
