// angular-pipes/esm5/array/drop.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var DropPipe = /** @class */ (function () {
    function DropPipe() {
    }
    DropPipe.prototype.transform = function (input, quantity) {
        if (!isArray(input)) {
            return input;
        }
        return input.slice(quantity || 1, input.lenth);
    };
DropPipe.ɵfac = function DropPipe_Factory(t) { return new (t || DropPipe)(); };
DropPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "drop", type: DropPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DropPipe, [{
        type: Pipe,
        args: [{
                name: 'drop'
            }]
    }], function () { return []; }, null); })();
    return DropPipe;
}());
export { DropPipe };
var NgDropPipeModule = /** @class */ (function () {
    function NgDropPipeModule() {
    }
NgDropPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDropPipeModule });
NgDropPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDropPipeModule_Factory(t) { return new (t || NgDropPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDropPipeModule, { declarations: [DropPipe], exports: [DropPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDropPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DropPipe],
                exports: [DropPipe]
            }]
    }], function () { return []; }, null); })();
    return NgDropPipeModule;
}());
export { NgDropPipeModule };
