// angular-pipes/esm5/array/every.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { every } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var EveryPipe = /** @class */ (function () {
    function EveryPipe() {
    }
    EveryPipe.prototype.transform = function (input, predicate) {
        return every(input, predicate);
    };
EveryPipe.ɵfac = function EveryPipe_Factory(t) { return new (t || EveryPipe)(); };
EveryPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "every", type: EveryPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EveryPipe, [{
        type: Pipe,
        args: [{
                name: 'every'
            }]
    }], function () { return []; }, null); })();
    return EveryPipe;
}());
export { EveryPipe };
var NgEveryPipeModule = /** @class */ (function () {
    function NgEveryPipeModule() {
    }
NgEveryPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEveryPipeModule });
NgEveryPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEveryPipeModule_Factory(t) { return new (t || NgEveryPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEveryPipeModule, { declarations: [EveryPipe], exports: [EveryPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEveryPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EveryPipe],
                exports: [EveryPipe]
            }]
    }], function () { return []; }, null); })();
    return NgEveryPipeModule;
}());
export { NgEveryPipeModule };
