// angular-pipes/esm5/array/first-or-default.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isFunction, getProperty } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var FirstOrDefaultPipe = /** @class */ (function () {
    function FirstOrDefaultPipe() {
    }
    FirstOrDefaultPipe_1 = FirstOrDefaultPipe;
    FirstOrDefaultPipe.find = function (input, predicate, defaultValue) {
        var found = false;
        var result;
        var i = -1;
        while (!found && ++i < input.length) {
            found = predicate(input[i], i, input);
        }
        if (found) {
            result = input[i];
        }
        if (typeof result === 'undefined' && typeof defaultValue !== 'undefined') {
            result = defaultValue;
        }
        return result;
    };
    FirstOrDefaultPipe.prototype.transform = function (input, predicate, defaultValue) {
        if (!isArray(input)) {
            return input;
        }
        if (isFunction(predicate)) {
            return FirstOrDefaultPipe_1.find(input, predicate, defaultValue);
        }
        else if (isArray(predicate)) {
            var _a = tslib_1.__read(predicate, 2), key_1 = _a[0], value_1 = _a[1];
            return FirstOrDefaultPipe_1.find(input, function (item) { return getProperty(item, key_1) === value_1; }, defaultValue);
        }
        else if (predicate) {
            return FirstOrDefaultPipe_1.find(input, function (item) { return item === predicate; }, defaultValue);
        }
        else {
            return input;
        }
    };
    var FirstOrDefaultPipe_1;
FirstOrDefaultPipe.ɵfac = function FirstOrDefaultPipe_Factory(t) { return new (t || FirstOrDefaultPipe)(); };
FirstOrDefaultPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "firstOrDefault", type: FirstOrDefaultPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FirstOrDefaultPipe, [{
        type: Pipe,
        args: [{
                name: 'firstOrDefault'
            }]
    }], function () { return []; }, null); })();
    return FirstOrDefaultPipe;
}());
export { FirstOrDefaultPipe };
var NgFirstOrDefaultPipeModule = /** @class */ (function () {
    function NgFirstOrDefaultPipeModule() {
    }
NgFirstOrDefaultPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgFirstOrDefaultPipeModule });
NgFirstOrDefaultPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgFirstOrDefaultPipeModule_Factory(t) { return new (t || NgFirstOrDefaultPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgFirstOrDefaultPipeModule, { declarations: [FirstOrDefaultPipe], exports: [FirstOrDefaultPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgFirstOrDefaultPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [FirstOrDefaultPipe],
                exports: [FirstOrDefaultPipe]
            }]
    }], function () { return []; }, null); })();
    return NgFirstOrDefaultPipeModule;
}());
export { NgFirstOrDefaultPipeModule };
