// angular-pipes/esm5/array/initial.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var InitialPipe = /** @class */ (function () {
    function InitialPipe() {
    }
    InitialPipe.prototype.transform = function (input) {
        if (!isArray(input)) {
            return input;
        }
        return input.slice(0, input.length - 1);
    };
InitialPipe.ɵfac = function InitialPipe_Factory(t) { return new (t || InitialPipe)(); };
InitialPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "initial", type: InitialPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InitialPipe, [{
        type: Pipe,
        args: [{
                name: 'initial'
            }]
    }], function () { return []; }, null); })();
    return InitialPipe;
}());
export { InitialPipe };
var NgInitialPipeModule = /** @class */ (function () {
    function NgInitialPipeModule() {
    }
NgInitialPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgInitialPipeModule });
NgInitialPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgInitialPipeModule_Factory(t) { return new (t || NgInitialPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgInitialPipeModule, { declarations: [InitialPipe], exports: [InitialPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgInitialPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [InitialPipe],
                exports: [InitialPipe]
            }]
    }], function () { return []; }, null); })();
    return NgInitialPipeModule;
}());
export { NgInitialPipeModule };
