// angular-pipes/esm5/array/join.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var JoinPipe = /** @class */ (function () {
    function JoinPipe() {
    }
    JoinPipe.prototype.transform = function (input, character) {
        if (character === void 0) { character = ''; }
        if (!isArray(input)) {
            return input;
        }
        return input.join(character);
    };
JoinPipe.ɵfac = function JoinPipe_Factory(t) { return new (t || JoinPipe)(); };
JoinPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "join", type: JoinPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(JoinPipe, [{
        type: Pipe,
        args: [{
                name: 'join'
            }]
    }], function () { return []; }, null); })();
    return JoinPipe;
}());
export { JoinPipe };
var NgJoinPipeModule = /** @class */ (function () {
    function NgJoinPipeModule() {
    }
NgJoinPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgJoinPipeModule });
NgJoinPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgJoinPipeModule_Factory(t) { return new (t || NgJoinPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgJoinPipeModule, { declarations: [JoinPipe], exports: [JoinPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgJoinPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [JoinPipe],
                exports: [JoinPipe]
            }]
    }], function () { return []; }, null); })();
    return NgJoinPipeModule;
}());
export { NgJoinPipeModule };
