// angular-pipes/esm5/array/map.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var MapPipe = /** @class */ (function () {
    function MapPipe() {
    }
    MapPipe.prototype.transform = function (input, fn) {
        if (!isArray(input) || !fn) {
            return input;
        }
        return input.map(fn);
    };
MapPipe.ɵfac = function MapPipe_Factory(t) { return new (t || MapPipe)(); };
MapPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "map", type: MapPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MapPipe, [{
        type: Pipe,
        args: [{
                name: 'map'
            }]
    }], function () { return []; }, null); })();
    return MapPipe;
}());
export { MapPipe };
var NgMapPipeModule = /** @class */ (function () {
    function NgMapPipeModule() {
    }
NgMapPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMapPipeModule });
NgMapPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMapPipeModule_Factory(t) { return new (t || NgMapPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMapPipeModule, { declarations: [MapPipe], exports: [MapPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMapPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MapPipe],
                exports: [MapPipe]
            }]
    }], function () { return []; }, null); })();
    return NgMapPipeModule;
}());
export { NgMapPipeModule };
