// angular-pipes/esm5/array/pluck.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, getProperty } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var PluckPipe = /** @class */ (function () {
    function PluckPipe() {
    }
    PluckPipe.prototype.transform = function (input, key) {
        if (!isArray(input) || !key) {
            return input;
        }
        return input.map(function (value) {
            return getProperty(value, key);
        });
    };
PluckPipe.ɵfac = function PluckPipe_Factory(t) { return new (t || PluckPipe)(); };
PluckPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "pluck", type: PluckPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PluckPipe, [{
        type: Pipe,
        args: [{
                name: 'pluck'
            }]
    }], function () { return []; }, null); })();
    return PluckPipe;
}());
export { PluckPipe };
var NgPluckPipeModule = /** @class */ (function () {
    function NgPluckPipeModule() {
    }
NgPluckPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgPluckPipeModule });
NgPluckPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgPluckPipeModule_Factory(t) { return new (t || NgPluckPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgPluckPipeModule, { declarations: [PluckPipe], exports: [PluckPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgPluckPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [PluckPipe],
                exports: [PluckPipe]
            }]
    }], function () { return []; }, null); })();
    return NgPluckPipeModule;
}());
export { NgPluckPipeModule };
