// angular-pipes/esm5/array/range.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var RangePipe = /** @class */ (function () {
    function RangePipe() {
    }
    RangePipe.prototype.transform = function (_input, size, start, step) {
        if (size === void 0) { size = 0; }
        if (start === void 0) { start = 1; }
        if (step === void 0) { step = 1; }
        var range = [];
        for (var length_1 = 0; length_1 < size; ++length_1) {
            range.push(start);
            start += step;
        }
        return range;
    };
RangePipe.ɵfac = function RangePipe_Factory(t) { return new (t || RangePipe)(); };
RangePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "range", type: RangePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RangePipe, [{
        type: Pipe,
        args: [{
                name: 'range'
            }]
    }], function () { return []; }, null); })();
    return RangePipe;
}());
export { RangePipe };
var NgRangePipeModule = /** @class */ (function () {
    function NgRangePipeModule() {
    }
NgRangePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRangePipeModule });
NgRangePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRangePipeModule_Factory(t) { return new (t || NgRangePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRangePipeModule, { declarations: [RangePipe], exports: [RangePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRangePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RangePipe],
                exports: [RangePipe]
            }]
    }], function () { return []; }, null); })();
    return NgRangePipeModule;
}());
export { NgRangePipeModule };
