// angular-pipes/esm5/array/reverse.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var ReversePipe = /** @class */ (function () {
    function ReversePipe() {
    }
    ReversePipe.prototype.transform = function (input) {
        if (!isArray(input)) {
            return input;
        }
        return tslib_1.__spread(input).reverse();
    };
ReversePipe.ɵfac = function ReversePipe_Factory(t) { return new (t || ReversePipe)(); };
ReversePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "reverse", type: ReversePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ReversePipe, [{
        type: Pipe,
        args: [{
                name: 'reverse'
            }]
    }], function () { return []; }, null); })();
    return ReversePipe;
}());
export { ReversePipe };
var NgReversePipeModule = /** @class */ (function () {
    function NgReversePipeModule() {
    }
NgReversePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgReversePipeModule });
NgReversePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgReversePipeModule_Factory(t) { return new (t || NgReversePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgReversePipeModule, { declarations: [ReversePipe], exports: [ReversePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgReversePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ReversePipe],
                exports: [ReversePipe]
            }]
    }], function () { return []; }, null); })();
    return NgReversePipeModule;
}());
export { NgReversePipeModule };
