// angular-pipes/esm5/array/shuffle.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { shuffle } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var ShufflePipe = /** @class */ (function () {
    function ShufflePipe() {
    }
    ShufflePipe.prototype.transform = function (input) {
        return shuffle(input);
    };
ShufflePipe.ɵfac = function ShufflePipe_Factory(t) { return new (t || ShufflePipe)(); };
ShufflePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "shuffle", type: ShufflePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ShufflePipe, [{
        type: Pipe,
        args: [{
                name: 'shuffle'
            }]
    }], function () { return []; }, null); })();
    return ShufflePipe;
}());
export { ShufflePipe };
var NgShufflePipeModule = /** @class */ (function () {
    function NgShufflePipeModule() {
    }
NgShufflePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgShufflePipeModule });
NgShufflePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgShufflePipeModule_Factory(t) { return new (t || NgShufflePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgShufflePipeModule, { declarations: [ShufflePipe], exports: [ShufflePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgShufflePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ShufflePipe],
                exports: [ShufflePipe]
            }]
    }], function () { return []; }, null); })();
    return NgShufflePipeModule;
}());
export { NgShufflePipeModule };
