// angular-pipes/esm5/array/some.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var SomePipe = /** @class */ (function () {
    function SomePipe() {
    }
    SomePipe.prototype.transform = function (input, predicate) {
        if (!isArray(input) || !predicate) {
            return input;
        }
        var result = false;
        var i = -1;
        while (++i < input.length && !result) {
            result = predicate(input[i], i, input);
        }
        return result;
    };
SomePipe.ɵfac = function SomePipe_Factory(t) { return new (t || SomePipe)(); };
SomePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "some", type: SomePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SomePipe, [{
        type: Pipe,
        args: [{
                name: 'some'
            }]
    }], function () { return []; }, null); })();
    return SomePipe;
}());
export { SomePipe };
var NgSomePipeModule = /** @class */ (function () {
    function NgSomePipeModule() {
    }
NgSomePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSomePipeModule });
NgSomePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSomePipeModule_Factory(t) { return new (t || NgSomePipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSomePipeModule, { declarations: [SomePipe], exports: [SomePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSomePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SomePipe],
                exports: [SomePipe]
            }]
    }], function () { return []; }, null); })();
    return NgSomePipeModule;
}());
export { NgSomePipeModule };
