// angular-pipes/esm5/array/tail.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var TailPipe = /** @class */ (function () {
    function TailPipe() {
    }
    TailPipe.prototype.transform = function (input) {
        if (!isArray(input)) {
            return input;
        }
        return input.slice(1, input.length);
    };
TailPipe.ɵfac = function TailPipe_Factory(t) { return new (t || TailPipe)(); };
TailPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "tail", type: TailPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TailPipe, [{
        type: Pipe,
        args: [{
                name: 'tail'
            }]
    }], function () { return []; }, null); })();
    return TailPipe;
}());
export { TailPipe };
var NgTailPipeModule = /** @class */ (function () {
    function NgTailPipeModule() {
    }
NgTailPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTailPipeModule });
NgTailPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTailPipeModule_Factory(t) { return new (t || NgTailPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTailPipeModule, { declarations: [TailPipe], exports: [TailPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTailPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TailPipe],
                exports: [TailPipe]
            }]
    }], function () { return []; }, null); })();
    return NgTailPipeModule;
}());
export { NgTailPipeModule };
