// angular-pipes/esm5/array/union.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var UnionPipe = /** @class */ (function () {
    function UnionPipe() {
    }
    UnionPipe.prototype.transform = function (a, b) {
        if ((!isArray(a) && !isDeepObject(a)) || !isArray(b)) {
            return [];
        }
        if (isDeepObject(a)) {
            var unwrapped = unwrapDeep(a);
            if (!isArray(unwrapped)) {
                return [];
            }
            return []
                .concat(unwrapped)
                .concat(b)
                .filter(function (value, index, input) { return deepIndexOf(input, value) === index; });
        }
        return []
            .concat(a)
            .concat(b)
            .filter(function (value, index, input) { return input.indexOf(value) === index; });
    };
UnionPipe.ɵfac = function UnionPipe_Factory(t) { return new (t || UnionPipe)(); };
UnionPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "union", type: UnionPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UnionPipe, [{
        type: Pipe,
        args: [{
                name: 'union'
            }]
    }], function () { return []; }, null); })();
    return UnionPipe;
}());
export { UnionPipe };
var NgUnionPipeModule = /** @class */ (function () {
    function NgUnionPipeModule() {
    }
NgUnionPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUnionPipeModule });
NgUnionPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUnionPipeModule_Factory(t) { return new (t || NgUnionPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUnionPipeModule, { declarations: [UnionPipe], exports: [UnionPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUnionPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [UnionPipe],
                exports: [UnionPipe]
            }]
    }], function () { return []; }, null); })();
    return NgUnionPipeModule;
}());
export { NgUnionPipeModule };
