// angular-pipes/esm5/array/uniq.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var UniqPipe = /** @class */ (function () {
    function UniqPipe() {
    }
    UniqPipe.prototype.transform = function (input) {
        if (!isArray(input) && !isDeepObject(input)) {
            return input;
        }
        if (isDeepObject(input)) {
            var unwrappedInput_1 = unwrapDeep(input);
            if (!isArray(unwrappedInput_1)) {
                return unwrappedInput_1;
            }
            return unwrappedInput_1.filter(function (value, index) { return deepIndexOf(unwrappedInput_1, value) === index; });
        }
        return input.filter(function (value, index) { return input.indexOf(value) === index; });
    };
UniqPipe.ɵfac = function UniqPipe_Factory(t) { return new (t || UniqPipe)(); };
UniqPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "uniq", type: UniqPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UniqPipe, [{
        type: Pipe,
        args: [{
                name: 'uniq'
            }]
    }], function () { return []; }, null); })();
    return UniqPipe;
}());
export { UniqPipe };
var NgUniqPipeModule = /** @class */ (function () {
    function NgUniqPipeModule() {
    }
NgUniqPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUniqPipeModule });
NgUniqPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUniqPipeModule_Factory(t) { return new (t || NgUniqPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUniqPipeModule, { declarations: [UniqPipe], exports: [UniqPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUniqPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [UniqPipe],
                exports: [UniqPipe]
            }]
    }], function () { return []; }, null); })();
    return NgUniqPipeModule;
}());
export { NgUniqPipeModule };
