// angular-pipes/esm5/array/without.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, isDeepObject, unwrapDeep, deepIndexOf } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var WithoutPipe = /** @class */ (function () {
    function WithoutPipe() {
    }
    WithoutPipe.prototype.transform = function (input) {
        var args = [];
        for (var _i = 1; _i < arguments.length; _i++) {
            args[_i - 1] = arguments[_i];
        }
        if (!isArray(input) && !isDeepObject(input)) {
            return input;
        }
        if (isDeepObject(input)) {
            var unwrappedInput = unwrapDeep(input);
            if (!isArray(unwrappedInput)) {
                return unwrappedInput;
            }
            return unwrappedInput.filter(function (value) { return deepIndexOf(args, value) === -1; });
        }
        return input.filter(function (value) { return args.indexOf(value) === -1; });
    };
WithoutPipe.ɵfac = function WithoutPipe_Factory(t) { return new (t || WithoutPipe)(); };
WithoutPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "without", type: WithoutPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WithoutPipe, [{
        type: Pipe,
        args: [{
                name: 'without'
            }]
    }], function () { return []; }, null); })();
    return WithoutPipe;
}());
export { WithoutPipe };
var NgWithoutPipeModule = /** @class */ (function () {
    function NgWithoutPipeModule() {
    }
NgWithoutPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgWithoutPipeModule });
NgWithoutPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgWithoutPipeModule_Factory(t) { return new (t || NgWithoutPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgWithoutPipeModule, { declarations: [WithoutPipe], exports: [WithoutPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgWithoutPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [WithoutPipe],
                exports: [WithoutPipe]
            }]
    }], function () { return []; }, null); })();
    return NgWithoutPipeModule;
}());
export { NgWithoutPipeModule };
