// angular-pipes/esm5/aggregate/max.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var MaxPipe = /** @class */ (function () {
    function MaxPipe() {
    }
    MaxPipe.prototype.transform = function (input) {
        if (!isArray(input)) {
            return input;
        }
        if (input.length === 0) {
            return undefined;
        }
        var max = input[0];
        input.forEach(function (value) {
            if (max < value) {
                max = value;
            }
        });
        return max;
    };
MaxPipe.ɵfac = function MaxPipe_Factory(t) { return new (t || MaxPipe)(); };
MaxPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "max", type: MaxPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaxPipe, [{
        type: Pipe,
        args: [{
                name: 'max'
            }]
    }], function () { return []; }, null); })();
    return MaxPipe;
}());
export { MaxPipe };
var NgMaxPipeModule = /** @class */ (function () {
    function NgMaxPipeModule() {
    }
NgMaxPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMaxPipeModule });
NgMaxPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMaxPipeModule_Factory(t) { return new (t || NgMaxPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMaxPipeModule, { declarations: [MaxPipe], exports: [MaxPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMaxPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MaxPipe],
                exports: [MaxPipe]
            }]
    }], function () { return []; }, null); })();
    return NgMaxPipeModule;
}());
export { NgMaxPipeModule };
