// angular-pipes/esm5/aggregate/mean.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, sum } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var MeanPipe = /** @class */ (function () {
    function MeanPipe() {
    }
    MeanPipe.prototype.transform = function (input) {
        if (!isArray(input)) {
            return input;
        }
        var count = input.length;
        if (count === 0) {
            return undefined;
        }
        return sum(input) / count;
    };
MeanPipe.ɵfac = function MeanPipe_Factory(t) { return new (t || MeanPipe)(); };
MeanPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "mean", type: MeanPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MeanPipe, [{
        type: Pipe,
        args: [{
                name: 'mean'
            }]
    }], function () { return []; }, null); })();
    return MeanPipe;
}());
export { MeanPipe };
var NgMeanPipeModule = /** @class */ (function () {
    function NgMeanPipeModule() {
    }
NgMeanPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgMeanPipeModule });
NgMeanPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgMeanPipeModule_Factory(t) { return new (t || NgMeanPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgMeanPipeModule, { declarations: [MeanPipe], exports: [MeanPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgMeanPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [MeanPipe],
                exports: [MeanPipe]
            }]
    }], function () { return []; }, null); })();
    return NgMeanPipeModule;
}());
export { NgMeanPipeModule };
