// angular-pipes/esm5/aggregate/sum.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isArray, sum } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
var SumPipe = /** @class */ (function () {
    function SumPipe() {
    }
    SumPipe.prototype.transform = function (input) {
        return !isArray(input) ? input : sum(input);
    };
SumPipe.ɵfac = function SumPipe_Factory(t) { return new (t || SumPipe)(); };
SumPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "sum", type: SumPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SumPipe, [{
        type: Pipe,
        args: [{ name: 'sum' }]
    }], function () { return []; }, null); })();
    return SumPipe;
}());
export { SumPipe };
var NgSumPipeModule = /** @class */ (function () {
    function NgSumPipeModule() {
    }
NgSumPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSumPipeModule });
NgSumPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSumPipeModule_Factory(t) { return new (t || NgSumPipeModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSumPipeModule, { declarations: [SumPipe], exports: [SumPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSumPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SumPipe],
                exports: [SumPipe]
            }]
    }], function () { return []; }, null); })();
    return NgSumPipeModule;
}());
export { NgSumPipeModule };
