// angular-pipes/esm2015/string/decode-uri-component.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let DecodeURIComponentPipe = class DecodeURIComponentPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return decodeURIComponent(input);
    }
};
DecodeURIComponentPipe.ɵfac = function DecodeURIComponentPipe_Factory(t) { return new (t || DecodeURIComponentPipe)(); };
DecodeURIComponentPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "decodeURIComponent", type: DecodeURIComponentPipe, pure: true });
export { DecodeURIComponentPipe };
let NgDecodeURIComponentPipeModule = class NgDecodeURIComponentPipeModule {
};
NgDecodeURIComponentPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDecodeURIComponentPipeModule });
NgDecodeURIComponentPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDecodeURIComponentPipeModule_Factory(t) { return new (t || NgDecodeURIComponentPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DecodeURIComponentPipe, [{
        type: Pipe,
        args: [{
                name: 'decodeURIComponent'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDecodeURIComponentPipeModule, { declarations: [DecodeURIComponentPipe], exports: [DecodeURIComponentPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDecodeURIComponentPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DecodeURIComponentPipe],
                exports: [DecodeURIComponentPipe]
            }]
    }], null, null); })();
export { NgDecodeURIComponentPipeModule };
