// angular-pipes/esm2015/string/decode-uri.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let DecodeURIPipe = class DecodeURIPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return decodeURI(input);
    }
};
DecodeURIPipe.ɵfac = function DecodeURIPipe_Factory(t) { return new (t || DecodeURIPipe)(); };
DecodeURIPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "decodeURI", type: DecodeURIPipe, pure: true });
export { DecodeURIPipe };
let NgDecodeURIPipeModule = class NgDecodeURIPipeModule {
};
NgDecodeURIPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgDecodeURIPipeModule });
NgDecodeURIPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgDecodeURIPipeModule_Factory(t) { return new (t || NgDecodeURIPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DecodeURIPipe, [{
        type: Pipe,
        args: [{
                name: 'decodeURI'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgDecodeURIPipeModule, { declarations: [DecodeURIPipe], exports: [DecodeURIPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgDecodeURIPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [DecodeURIPipe],
                exports: [DecodeURIPipe]
            }]
    }], null, null); })();
export { NgDecodeURIPipeModule };
