// angular-pipes/esm2015/string/encode-uri-component.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let EncodeURIComponentPipe = class EncodeURIComponentPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return encodeURIComponent(input);
    }
};
EncodeURIComponentPipe.ɵfac = function EncodeURIComponentPipe_Factory(t) { return new (t || EncodeURIComponentPipe)(); };
EncodeURIComponentPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "encodeURIComponent", type: EncodeURIComponentPipe, pure: true });
export { EncodeURIComponentPipe };
let NgEncodeURIComponentPipeModule = class NgEncodeURIComponentPipeModule {
};
NgEncodeURIComponentPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEncodeURIComponentPipeModule });
NgEncodeURIComponentPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEncodeURIComponentPipeModule_Factory(t) { return new (t || NgEncodeURIComponentPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EncodeURIComponentPipe, [{
        type: Pipe,
        args: [{
                name: 'encodeURIComponent'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEncodeURIComponentPipeModule, { declarations: [EncodeURIComponentPipe], exports: [EncodeURIComponentPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEncodeURIComponentPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EncodeURIComponentPipe],
                exports: [EncodeURIComponentPipe]
            }]
    }], null, null); })();
export { NgEncodeURIComponentPipeModule };
