// angular-pipes/esm2015/string/encode-uri.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let EncodeURIPipe = class EncodeURIPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return encodeURI(input);
    }
};
EncodeURIPipe.ɵfac = function EncodeURIPipe_Factory(t) { return new (t || EncodeURIPipe)(); };
EncodeURIPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "encodeURI", type: EncodeURIPipe, pure: true });
export { EncodeURIPipe };
let NgEncodeURIPipeModule = class NgEncodeURIPipeModule {
};
NgEncodeURIPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgEncodeURIPipeModule });
NgEncodeURIPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgEncodeURIPipeModule_Factory(t) { return new (t || NgEncodeURIPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EncodeURIPipe, [{
        type: Pipe,
        args: [{
                name: 'encodeURI'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgEncodeURIPipeModule, { declarations: [EncodeURIPipe], exports: [EncodeURIPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgEncodeURIPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [EncodeURIPipe],
                exports: [EncodeURIPipe]
            }]
    }], null, null); })();
export { NgEncodeURIPipeModule };
