// angular-pipes/esm2015/string/left-pad.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { leftPad, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let LeftPadPipe = class LeftPadPipe {
    transform(input, length = 0, character = ' ') {
        if (!isString(input)) {
            return input;
        }
        return leftPad(input, length, character);
    }
};
LeftPadPipe.ɵfac = function LeftPadPipe_Factory(t) { return new (t || LeftPadPipe)(); };
LeftPadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "leftpad", type: LeftPadPipe, pure: true });
export { LeftPadPipe };
let NgLeftPadPipeModule = class NgLeftPadPipeModule {
};
NgLeftPadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgLeftPadPipeModule });
NgLeftPadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgLeftPadPipeModule_Factory(t) { return new (t || NgLeftPadPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LeftPadPipe, [{
        type: Pipe,
        args: [{
                name: 'leftpad'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgLeftPadPipeModule, { declarations: [LeftPadPipe], exports: [LeftPadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgLeftPadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [LeftPadPipe],
                exports: [LeftPadPipe]
            }]
    }], null, null); })();
export { NgLeftPadPipeModule };
