// angular-pipes/esm2015/string/newlines.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let NewlinesPipe = class NewlinesPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return input.replace(/(?:\r\n|\r|\n)/g, '<br />');
    }
};
NewlinesPipe.ɵfac = function NewlinesPipe_Factory(t) { return new (t || NewlinesPipe)(); };
NewlinesPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "newlines", type: NewlinesPipe, pure: true });
export { NewlinesPipe };
let NgNewlinesPipeModule = class NgNewlinesPipeModule {
};
NgNewlinesPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgNewlinesPipeModule });
NgNewlinesPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgNewlinesPipeModule_Factory(t) { return new (t || NgNewlinesPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NewlinesPipe, [{
        type: Pipe,
        args: [{
                name: 'newlines'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgNewlinesPipeModule, { declarations: [NewlinesPipe], exports: [NewlinesPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgNewlinesPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [NewlinesPipe],
                exports: [NewlinesPipe]
            }]
    }], null, null); })();
export { NgNewlinesPipeModule };
