// angular-pipes/esm2015/string/pad.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { pad, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let PadPipe = class PadPipe {
    transform(input, length = 0, character = ' ') {
        if (!isString(input)) {
            return input;
        }
        return pad(input, length, character);
    }
};
PadPipe.ɵfac = function PadPipe_Factory(t) { return new (t || PadPipe)(); };
PadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "pad", type: PadPipe, pure: true });
export { PadPipe };
let NgPadPipeModule = class NgPadPipeModule {
};
NgPadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgPadPipeModule });
NgPadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgPadPipeModule_Factory(t) { return new (t || NgPadPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PadPipe, [{
        type: Pipe,
        args: [{
                name: 'pad'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgPadPipeModule, { declarations: [PadPipe], exports: [PadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgPadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [PadPipe],
                exports: [PadPipe]
            }]
    }], null, null); })();
export { NgPadPipeModule };
