// angular-pipes/esm2015/string/replace.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let ReplacePipe = class ReplacePipe {
    transform(input, pattern, replacement) {
        if (!isString(input) || isUndefined(pattern) || isUndefined(replacement)) {
            return input;
        }
        return input.replace(pattern, replacement);
    }
};
ReplacePipe.ɵfac = function ReplacePipe_Factory(t) { return new (t || ReplacePipe)(); };
ReplacePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "replace", type: ReplacePipe, pure: true });
export { ReplacePipe };
let NgReplacePipeModule = class NgReplacePipeModule {
};
NgReplacePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgReplacePipeModule });
NgReplacePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgReplacePipeModule_Factory(t) { return new (t || NgReplacePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ReplacePipe, [{
        type: Pipe,
        args: [{
                name: 'replace'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgReplacePipeModule, { declarations: [ReplacePipe], exports: [ReplacePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgReplacePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ReplacePipe],
                exports: [ReplacePipe]
            }]
    }], null, null); })();
export { NgReplacePipeModule };
