// angular-pipes/esm2015/string/reverse-str.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let ReverseStrPipe = class ReverseStrPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return Array.from(input)
            .reverse()
            .join('');
    }
};
ReverseStrPipe.ɵfac = function ReverseStrPipe_Factory(t) { return new (t || ReverseStrPipe)(); };
ReverseStrPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "reverseStr", type: ReverseStrPipe, pure: true });
export { ReverseStrPipe };
let NgReverseStrPipeModule = class NgReverseStrPipeModule {
};
NgReverseStrPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgReverseStrPipeModule });
NgReverseStrPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgReverseStrPipeModule_Factory(t) { return new (t || NgReverseStrPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ReverseStrPipe, [{
        type: Pipe,
        args: [{
                name: 'reverseStr'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgReverseStrPipeModule, { declarations: [ReverseStrPipe], exports: [ReverseStrPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgReverseStrPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [ReverseStrPipe],
                exports: [ReverseStrPipe]
            }]
    }], null, null); })();
export { NgReverseStrPipeModule };
