// angular-pipes/esm2015/string/right-pad.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { rightPad, isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let RightPadPipe = class RightPadPipe {
    transform(input, length = 0, character = ' ') {
        if (!isString(input)) {
            return input;
        }
        return rightPad(input, length, character);
    }
};
RightPadPipe.ɵfac = function RightPadPipe_Factory(t) { return new (t || RightPadPipe)(); };
RightPadPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "rightpad", type: RightPadPipe, pure: true });
export { RightPadPipe };
let NgRightPadPipeModule = class NgRightPadPipeModule {
};
NgRightPadPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgRightPadPipeModule });
NgRightPadPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgRightPadPipeModule_Factory(t) { return new (t || NgRightPadPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RightPadPipe, [{
        type: Pipe,
        args: [{
                name: 'rightpad'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgRightPadPipeModule, { declarations: [RightPadPipe], exports: [RightPadPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgRightPadPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [RightPadPipe],
                exports: [RightPadPipe]
            }]
    }], null, null); })();
export { NgRightPadPipeModule };
