// angular-pipes/esm2015/string/slugify.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let SlugifyPipe = class SlugifyPipe {
    transform(input) {
        if (!isString(input)) {
            return input;
        }
        return input
            .toLowerCase()
            .trim()
            .replace(/[^a-z0-9\s-]/g, ' ')
            .replace(/[\s-]+/g, '-');
    }
};
SlugifyPipe.ɵfac = function SlugifyPipe_Factory(t) { return new (t || SlugifyPipe)(); };
SlugifyPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "slugify", type: SlugifyPipe, pure: true });
export { SlugifyPipe };
let NgSlugifyPipeModule = class NgSlugifyPipeModule {
};
NgSlugifyPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgSlugifyPipeModule });
NgSlugifyPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgSlugifyPipeModule_Factory(t) { return new (t || NgSlugifyPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlugifyPipe, [{
        type: Pipe,
        args: [{ name: 'slugify' }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgSlugifyPipeModule, { declarations: [SlugifyPipe], exports: [SlugifyPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgSlugifyPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [SlugifyPipe],
                exports: [SlugifyPipe]
            }]
    }], null, null); })();
export { NgSlugifyPipeModule };
