// angular-pipes/esm2015/string/template.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TemplatePipe = class TemplatePipe {
    transform(input, ...args) {
        if (!isString(input) || args.length === 0) {
            return input;
        }
        let template = input;
        for (let i = 0; i < args.length; ++i) {
            template = template.replace(`$${i + 1}`, args[i]);
        }
        return template;
    }
};
TemplatePipe.ɵfac = function TemplatePipe_Factory(t) { return new (t || TemplatePipe)(); };
TemplatePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "template", type: TemplatePipe, pure: true });
export { TemplatePipe };
let NgTemplatePipeModule = class NgTemplatePipeModule {
};
NgTemplatePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTemplatePipeModule });
NgTemplatePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTemplatePipeModule_Factory(t) { return new (t || NgTemplatePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TemplatePipe, [{
        type: Pipe,
        args: [{
                name: 'template'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTemplatePipeModule, { declarations: [TemplatePipe], exports: [TemplatePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTemplatePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TemplatePipe],
                exports: [TemplatePipe]
            }]
    }], null, null); })();
export { NgTemplatePipeModule };
