// angular-pipes/esm2015/string/test.pipe.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TestPipe = class TestPipe {
    transform(input, pattern, flag) {
        if (!isString(input) || !pattern) {
            return input;
        }
        const regexp = pattern instanceof RegExp ? pattern : new RegExp(pattern, flag);
        return regexp.test(input);
    }
};
TestPipe.ɵfac = function TestPipe_Factory(t) { return new (t || TestPipe)(); };
TestPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "test", type: TestPipe, pure: true });
export { TestPipe };
let NgTestPipeModule = class NgTestPipeModule {
};
NgTestPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTestPipeModule });
NgTestPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTestPipeModule_Factory(t) { return new (t || NgTestPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TestPipe, [{
        type: Pipe,
        args: [{
                name: 'test'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTestPipeModule, { declarations: [TestPipe], exports: [TestPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTestPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TestPipe],
                exports: [TestPipe]
            }]
    }], null, null); })();
export { NgTestPipeModule };
