// angular-pipes/esm2015/string/truncate.pipe.js
// Inspired from https://github.com/a8m/angular-filter/blob/master/src/_filter/string/truncate.js
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let TruncatePipe = class TruncatePipe {
    transform(input, length, suffix, preserve) {
        if (!isString(input)) {
            return input;
        }
        length = isUndefined(length) ? input.length : length;
        if (input.length <= length) {
            return input;
        }
        preserve = preserve || false;
        suffix = suffix || '';
        let index = length;
        if (preserve) {
            if (input.indexOf(' ', length) === -1) {
                index = input.length;
            }
            else {
                index = input.indexOf(' ', length);
            }
        }
        return input.substring(0, index) + suffix;
    }
};
TruncatePipe.ɵfac = function TruncatePipe_Factory(t) { return new (t || TruncatePipe)(); };
TruncatePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "truncate", type: TruncatePipe, pure: true });
export { TruncatePipe };
let NgTruncatePipeModule = class NgTruncatePipeModule {
};
NgTruncatePipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgTruncatePipeModule });
NgTruncatePipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgTruncatePipeModule_Factory(t) { return new (t || NgTruncatePipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TruncatePipe, [{
        type: Pipe,
        args: [{
                name: 'truncate'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgTruncatePipeModule, { declarations: [TruncatePipe], exports: [TruncatePipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgTruncatePipeModule, [{
        type: NgModule,
        args: [{
                declarations: [TruncatePipe],
                exports: [TruncatePipe]
            }]
    }], null, null); })();
export { NgTruncatePipeModule };
