// angular-pipes/esm2015/string/wrap.pipe.js
// idea from https://github.com/a8m/angular-filter
import * as tslib_1 from "tslib";
import { Pipe, NgModule } from '@angular/core';
import { isString, isUndefined } from '../utils/utils';
import * as ɵngcc0 from '@angular/core';
let WrapPipe = class WrapPipe {
    transform(input, wrap, ends) {
        return isString(input) && !isUndefined(wrap) ? [wrap, input, ends || wrap].join('') : input;
    }
};
WrapPipe.ɵfac = function WrapPipe_Factory(t) { return new (t || WrapPipe)(); };
WrapPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "wrap", type: WrapPipe, pure: true });
export { WrapPipe };
let NgWrapPipeModule = class NgWrapPipeModule {
};
NgWrapPipeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgWrapPipeModule });
NgWrapPipeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgWrapPipeModule_Factory(t) { return new (t || NgWrapPipeModule)(); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WrapPipe, [{
        type: Pipe,
        args: [{
                name: 'wrap'
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgWrapPipeModule, { declarations: [WrapPipe], exports: [WrapPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgWrapPipeModule, [{
        type: NgModule,
        args: [{
                declarations: [WrapPipe],
                exports: [WrapPipe]
            }]
    }], null, null); })();
export { NgWrapPipeModule };
